// @aws-sdk/client-sesv2/dist-cjs/schemas/schemas_0.js
const _A = "Attachment";
const _AA = "ArchiveArn";
const _AAp = "ApplicationArn";
const _ACEA = "AdditionalContactEmailAddress";
const _ACEAd = "AdditionalContactEmailAddresses";
const _AD = "AccountDetails";
const _ADt = "AttributesData";
const _AE = "AdminEmail";
const _AEE = "AlreadyExistsException";
const _AL = "AttachmentList";
const _AO = "ArchivingOptions";
const _AS = "AccountStatus";
const _ASD = "ActiveSubscribedDomains";
const _ASE = "AccountSuspendedException";
const _ASMS = "AwsSesManagedStatus";
const _AT = "AssociatedTimestamp";
const _AWE = "AutoWarmupEnabled";
const _Ag = "Aggregation";
const _At = "Attachments";
const _Att = "Attributes";
const _B = "Body";
const _BA = "BccAddresses";
const _BE = "BlacklistEntry";
const _BEC = "BulkEmailContent";
const _BEE = "BulkEmailEntry";
const _BEEL = "BulkEmailEntryList";
const _BEER = "BulkEmailEntryResult";
const _BEERL = "BulkEmailEntryResultList";
const _BEERu = "BulkEmailEntryResults";
const _BEEu = "BulkEmailEntries";
const _BEl = "BlacklistEntries";
const _BGMD = "BatchGetMetricData";
const _BGMDQ = "BatchGetMetricDataQuery";
const _BGMDQa = "BatchGetMetricDataQueries";
const _BGMDR = "BatchGetMetricDataRequest";
const _BGMDRa = "BatchGetMetricDataResponse";
const _BIN = "BlacklistItemNames";
const _BOMF = "BehaviorOnMxFailure";
const _BR = "BlacklistReport";
const _BRE = "BadRequestException";
const _BST = "BounceSubType";
const _BT = "BounceType";
const _Bo = "Bounce";
const _C = "Complaint";
const _CA = "CcAddresses";
const _CC = "CreateContact";
const _CCL = "CreateContactList";
const _CCLR = "CreateContactListRequest";
const _CCLRr = "CreateContactListResponse";
const _CCR = "CreateContactRequest";
const _CCRr = "CreateContactResponse";
const _CCS = "CreateConfigurationSet";
const _CCSED = "CreateConfigurationSetEventDestination";
const _CCSEDR = "CreateConfigurationSetEventDestinationRequest";
const _CCSEDRr = "CreateConfigurationSetEventDestinationResponse";
const _CCSR = "CreateConfigurationSetRequest";
const _CCSRr = "CreateConfigurationSetResponse";
const _CCVET = "CreateCustomVerificationEmailTemplate";
const _CCVETR = "CreateCustomVerificationEmailTemplateRequest";
const _CCVETRr = "CreateCustomVerificationEmailTemplateResponse";
const _CD = "ContentDisposition";
const _CDIP = "CreateDedicatedIpPool";
const _CDIPR = "CreateDedicatedIpPoolRequest";
const _CDIPRr = "CreateDedicatedIpPoolResponse";
const _CDTR = "CreateDeliverabilityTestReport";
const _CDTRR = "CreateDeliverabilityTestReportRequest";
const _CDTRRr = "CreateDeliverabilityTestReportResponse";
const _CDo = "ContentDescription";
const _CDr = "CreateDate";
const _CE = "ConflictException";
const _CEI = "CreateEmailIdentity";
const _CEIP = "CreateEmailIdentityPolicy";
const _CEIPR = "CreateEmailIdentityPolicyRequest";
const _CEIPRr = "CreateEmailIdentityPolicyResponse";
const _CEIR = "CreateEmailIdentityRequest";
const _CEIRr = "CreateEmailIdentityResponse";
const _CEJ = "CancelExportJob";
const _CEJR = "CancelExportJobRequest";
const _CEJRa = "CancelExportJobResponse";
const _CEJRr = "CreateExportJobRequest";
const _CEJRre = "CreateExportJobResponse";
const _CEJr = "CreateExportJob";
const _CET = "CreateEmailTemplate";
const _CETR = "CreateEmailTemplateRequest";
const _CETRr = "CreateEmailTemplateResponse";
const _CFT = "ComplaintFeedbackType";
const _CI = "ContentId";
const _CIJ = "CreateImportJob";
const _CIJR = "CreateImportJobRequest";
const _CIJRr = "CreateImportJobResponse";
const _CIa = "CampaignId";
const _CIas = "CaseId";
const _CL = "ContactLanguage";
const _CLD = "ContactListDestination";
const _CLIA = "ContactListImportAction";
const _CLN = "ContactListName";
const _CLo = "ContactList";
const _CLon = "ContactLists";
const _CME = "ConcurrentModificationException";
const _CMRE = "CreateMultiRegionEndpoint";
const _CMRER = "CreateMultiRegionEndpointRequest";
const _CMRERr = "CreateMultiRegionEndpointResponse";
const _CMS = "CustomerManagedStatus";
const _CRD = "CustomRedirectDomain";
const _CS = "ConfigurationSets";
const _CSKL = "CurrentSigningKeyLength";
const _CSN = "ConfigurationSetName";
const _CST = "ComplaintSubType";
const _CT = "ContentType";
const _CTE = "ContentTransferEncoding";
const _CTEo = "ConditionThresholdEnabled";
const _CTR = "CreateTenantRequest";
const _CTRA = "CreateTenantResourceAssociation";
const _CTRAR = "CreateTenantResourceAssociationRequest";
const _CTRARr = "CreateTenantResourceAssociationResponse";
const _CTRr = "CreateTenantResponse";
const _CTo = "CompletedTimestamp";
const _CTon = "ConditionThreshold";
const _CTr = "CreatedTimestamp";
const _CTre = "CreateTenant";
const _CV = "ConfidenceVerdict";
const _CVET = "CustomVerificationEmailTemplates";
const _CVETL = "CustomVerificationEmailTemplatesList";
const _CVETM = "CustomVerificationEmailTemplateMetadata";
const _CVT = "ConfidenceVerdictThreshold";
const _CWD = "CloudWatchDestination";
const _CWDC = "CloudWatchDimensionConfiguration";
const _CWDCl = "CloudWatchDimensionConfigurations";
const _Ca = "Cause";
const _Ch = "Charset";
const _Co = "Contact";
const _Cod = "Code";
const _Con = "Content";
const _Cont = "Contacts";
const _D = "Dimensions";
const _DA = "DkimAttributes";
const _DAa = "DashboardAttributes";
const _DC = "DiagnosticCode";
const _DCL = "DeleteContactList";
const _DCLR = "DeleteContactListRequest";
const _DCLRe = "DeleteContactListResponse";
const _DCR = "DeleteContactRequest";
const _DCRe = "DeleteContactResponse";
const _DCS = "DeleteConfigurationSet";
const _DCSED = "DeleteConfigurationSetEventDestination";
const _DCSEDR = "DeleteConfigurationSetEventDestinationRequest";
const _DCSEDRe = "DeleteConfigurationSetEventDestinationResponse";
const _DCSR = "DeleteConfigurationSetRequest";
const _DCSRe = "DeleteConfigurationSetResponse";
const _DCVET = "DeleteCustomVerificationEmailTemplate";
const _DCVETR = "DeleteCustomVerificationEmailTemplateRequest";
const _DCVETRe = "DeleteCustomVerificationEmailTemplateResponse";
const _DCe = "DefaultContent";
const _DCel = "DeleteContact";
const _DCi = "DimensionConfigurations";
const _DDC = "DomainDeliverabilityCampaign";
const _DDCL = "DomainDeliverabilityCampaignList";
const _DDCo = "DomainDeliverabilityCampaigns";
const _DDIP = "DeleteDedicatedIpPool";
const _DDIPR = "DeleteDedicatedIpPoolRequest";
const _DDIPRe = "DeleteDedicatedIpPoolResponse";
const _DDTO = "DomainDeliverabilityTrackingOption";
const _DDTOo = "DomainDeliverabilityTrackingOptions";
const _DDV = "DefaultDimensionValue";
const _DE = "DashboardEnabled";
const _DEI = "DeleteEmailIdentity";
const _DEIP = "DeleteEmailIdentityPolicy";
const _DEIPR = "DeleteEmailIdentityPolicyRequest";
const _DEIPRe = "DeleteEmailIdentityPolicyResponse";
const _DEIR = "DeleteEmailIdentityRequest";
const _DEIRe = "DeleteEmailIdentityResponse";
const _DET = "DefaultEmailTags";
const _DETR = "DeleteEmailTemplateRequest";
const _DETRe = "DeleteEmailTemplateResponse";
const _DETe = "DeleteEmailTemplate";
const _DF = "DataFormat";
const _DI = "DedicatedIp";
const _DIAWE = "DedicatedIpAutoWarmupEnabled";
const _DIL = "DedicatedIpList";
const _DIP = "DomainIspPlacements";
const _DIPe = "DedicatedIpPool";
const _DIPed = "DedicatedIpPools";
const _DIPo = "DomainIspPlacement";
const _DIe = "DedicatedIps";
const _DMRE = "DeleteMultiRegionEndpoint";
const _DMRER = "DeleteMultiRegionEndpointRequest";
const _DMRERe = "DeleteMultiRegionEndpointResponse";
const _DN = "DimensionName";
const _DNi = "DisplayName";
const _DO = "DeliveryOptions";
const _DOa = "DashboardOptions";
const _DP = "DkimPercentage";
const _DPN = "DestinationPoolName";
const _DR = "DeleteRate";
const _DS = "DkimStatus";
const _DSA = "DkimSigningAttributes";
const _DSAO = "DomainSigningAttributesOrigin";
const _DSAe = "DeliveryStreamArn";
const _DSD = "DeleteSuppressedDestination";
const _DSDR = "DeleteSuppressedDestinationRequest";
const _DSDRe = "DeleteSuppressedDestinationResponse";
const _DSPK = "DomainSigningPrivateKey";
const _DSS = "DomainSigningSelector";
const _DSSe = "DefaultSubscriptionStatus";
const _DT = "DkimTokens";
const _DTR = "DeleteTenantRequest";
const _DTRA = "DeleteTenantResourceAssociation";
const _DTRAR = "DeleteTenantResourceAssociationRequest";
const _DTRARe = "DeleteTenantResourceAssociationResponse";
const _DTRe = "DeleteTenantResponse";
const _DTRel = "DeliverabilityTestReport";
const _DTReli = "DeliverabilityTestReports";
const _DTS = "DeliverabilityTestStatus";
const _DTe = "DeleteTenant";
const _DV = "DailyVolume";
const _DVS = "DimensionValueSource";
const _DVa = "DailyVolumes";
const _Da = "Data";
const _De = "Description";
const _Des = "Destination";
const _Det = "Details";
const _Do = "Domain";
const _E = "Errors";
const _EA = "EmailAddress";
const _EAFL = "EmailAddressFilterList";
const _EAIME = "EmailAddressInsightsMailboxEvaluations";
const _EAIV = "EmailAddressInsightsVerdict";
const _EBA = "EventBusArn";
const _EBD = "EventBridgeDestination";
const _EC = "EmailContent";
const _ED = "EndDate";
const _EDD = "EventDestinationDefinition";
const _EDN = "EventDestinationName";
const _EDS = "ExportDataSource";
const _EDv = "EventDestination";
const _EDve = "EventDetails";
const _EDven = "EventDestinations";
const _EDx = "ExportDestination";
const _EDxp = "ExportDimensions";
const _EFE = "EmailForwardingEnabled";
const _EI = "EmailIdentity";
const _EIL = "EmailInsightsList";
const _EIm = "EmailInsights";
const _EIma = "EmailIdentities";
const _EIn = "EndpointId";
const _EJ = "ExportJobs";
const _EJS = "ExportJobSummary";
const _EJSL = "ExportJobSummaryList";
const _EM = "EngagementMetrics";
const _EMr = "ErrorMessage";
const _EMx = "ExportMetric";
const _EMxp = "ExportMetrics";
const _EN = "EndpointName";
const _ERC = "ExportedRecordsCount";
const _ES = "EmailSubject";
const _ESFL = "EmailSubjectFilterList";
const _EST = "ExportSourceType";
const _ESn = "EnforcementStatus";
const _ESx = "ExportStatistics";
const _ET = "EmailTags";
const _ETC = "EmailTemplateContent";
const _ETM = "EmailTemplateMetadata";
const _ETML = "EmailTemplateMetadataList";
const _ETr = "ErrorType";
const _En = "Enabled";
const _Er = "Error";
const _Es = "Esps";
const _Ev = "Events";
const _Eva = "Evaluations";
const _Ex = "Exclude";
const _F = "Filter";
const _FA = "FromAddress";
const _FEA = "FromEmailAddress";
const _FEAIA = "FromEmailAddressIdentityArn";
const _FFEA = "FeedbackForwardingEmailAddress";
const _FFEAIA = "FeedbackForwardingEmailAddressIdentityArn";
const _FFS = "FeedbackForwardingStatus";
const _FI = "FailureInfo";
const _FIe = "FeedbackId";
const _FN = "FileName";
const _FRC = "FailedRecordsCount";
const _FRSU = "FailedRecordsS3Url";
const _FRURL = "FailureRedirectionURL";
const _FS = "FilteredStatus";
const _FSDT = "FirstSeenDateTime";
const _G = "Global";
const _GA = "GuardianAttributes";
const _GAR = "GetAccountRequest";
const _GARe = "GetAccountResponse";
const _GAe = "GetAccount";
const _GBR = "GetBlacklistReports";
const _GBRR = "GetBlacklistReportsRequest";
const _GBRRe = "GetBlacklistReportsResponse";
const _GC = "GetContact";
const _GCL = "GetContactList";
const _GCLR = "GetContactListRequest";
const _GCLRe = "GetContactListResponse";
const _GCR = "GetContactRequest";
const _GCRe = "GetContactResponse";
const _GCS = "GetConfigurationSet";
const _GCSED = "GetConfigurationSetEventDestinations";
const _GCSEDR = "GetConfigurationSetEventDestinationsRequest";
const _GCSEDRe = "GetConfigurationSetEventDestinationsResponse";
const _GCSR = "GetConfigurationSetRequest";
const _GCSRe = "GetConfigurationSetResponse";
const _GCVET = "GetCustomVerificationEmailTemplate";
const _GCVETR = "GetCustomVerificationEmailTemplateRequest";
const _GCVETRe = "GetCustomVerificationEmailTemplateResponse";
const _GDDC = "GetDomainDeliverabilityCampaign";
const _GDDCR = "GetDomainDeliverabilityCampaignRequest";
const _GDDCRe = "GetDomainDeliverabilityCampaignResponse";
const _GDDO = "GetDeliverabilityDashboardOptions";
const _GDDOR = "GetDeliverabilityDashboardOptionsRequest";
const _GDDORe = "GetDeliverabilityDashboardOptionsResponse";
const _GDI = "GetDedicatedIp";
const _GDIP = "GetDedicatedIpPool";
const _GDIPR = "GetDedicatedIpPoolRequest";
const _GDIPRe = "GetDedicatedIpPoolResponse";
const _GDIR = "GetDedicatedIpRequest";
const _GDIRe = "GetDedicatedIpResponse";
const _GDIRet = "GetDedicatedIpsRequest";
const _GDIRete = "GetDedicatedIpsResponse";
const _GDIe = "GetDedicatedIps";
const _GDSR = "GetDomainStatisticsReport";
const _GDSRR = "GetDomainStatisticsReportRequest";
const _GDSRRe = "GetDomainStatisticsReportResponse";
const _GDTR = "GetDeliverabilityTestReport";
const _GDTRR = "GetDeliverabilityTestReportRequest";
const _GDTRRe = "GetDeliverabilityTestReportResponse";
const _GEAI = "GetEmailAddressInsights";
const _GEAIR = "GetEmailAddressInsightsRequest";
const _GEAIRe = "GetEmailAddressInsightsResponse";
const _GEI = "GetEmailIdentity";
const _GEIP = "GetEmailIdentityPolicies";
const _GEIPR = "GetEmailIdentityPoliciesRequest";
const _GEIPRe = "GetEmailIdentityPoliciesResponse";
const _GEIR = "GetEmailIdentityRequest";
const _GEIRe = "GetEmailIdentityResponse";
const _GEJ = "GetExportJob";
const _GEJR = "GetExportJobRequest";
const _GEJRe = "GetExportJobResponse";
const _GET = "GetEmailTemplate";
const _GETR = "GetEmailTemplateRequest";
const _GETRe = "GetEmailTemplateResponse";
const _GIJ = "GetImportJob";
const _GIJR = "GetImportJobRequest";
const _GIJRe = "GetImportJobResponse";
const _GMI = "GetMessageInsights";
const _GMIR = "GetMessageInsightsRequest";
const _GMIRe = "GetMessageInsightsResponse";
const _GMRE = "GetMultiRegionEndpoint";
const _GMRER = "GetMultiRegionEndpointRequest";
const _GMRERe = "GetMultiRegionEndpointResponse";
const _GO = "GuardianOptions";
const _GRE = "GetReputationEntity";
const _GRER = "GetReputationEntityRequest";
const _GRERe = "GetReputationEntityResponse";
const _GSD = "GetSuppressedDestination";
const _GSDR = "GetSuppressedDestinationRequest";
const _GSDRe = "GetSuppressedDestinationResponse";
const _GT = "GetTenant";
const _GTR = "GetTenantRequest";
const _GTRe = "GetTenantResponse";
const _H = "Html";
const _HP = "HttpsPolicy";
const _HVDR = "HasValidDnsRecords";
const _HVS = "HasValidSyntax";
const _He = "Headers";
const _I = "Id";
const _IC = "InboxCount";
const _ID = "ImportDestination";
const _IDS = "ImportDataSource";
const _IDT = "ImportDestinationType";
const _IDs = "IsDisposable";
const _IE = "InsightsEvent";
const _IEA = "InsightsEmailAddress";
const _IEn = "InsightsEvents";
const _II = "IdentityInfo";
const _IIL = "IdentityInfoList";
const _IJ = "ImportJobs";
const _IJS = "ImportJobSummary";
const _IJSL = "ImportJobSummaryList";
const _IN = "IspName";
const _INTE = "InvalidNextTokenException";
const _INd = "IdentityName";
const _IP = "InboxPercentage";
const _IPTO = "InboxPlacementTrackingOption";
const _IPs = "IspPlacements";
const _IPsp = "IspPlacement";
const _IRA = "IsRoleAddress";
const _IRAa = "IamRoleArn";
const _IRC = "InboxRawCount";
const _IRI = "IsRandomInput";
const _ISEE = "InternalServiceErrorException";
const _IT = "IdentityType";
const _IU = "ImageUrl";
const _IV = "IsValid";
const _Im = "Impact";
const _In = "Insights";
const _Inc = "Include";
const _Ip = "Ip";
const _Is = "Isp";
const _JI = "JobId";
const _JS = "JobStatus";
const _K = "Key";
const _KFD = "KinesisFirehoseDestination";
const _LC = "ListContacts";
const _LCF = "ListContactsFilter";
const _LCL = "ListContactLists";
const _LCLR = "ListContactListsRequest";
const _LCLRi = "ListContactListsResponse";
const _LCR = "ListContactsRequest";
const _LCRi = "ListContactsResponse";
const _LCS = "ListConfigurationSets";
const _LCSR = "ListConfigurationSetsRequest";
const _LCSRi = "ListConfigurationSetsResponse";
const _LCT = "LastCheckedTimestamp";
const _LCVET = "ListCustomVerificationEmailTemplates";
const _LCVETR = "ListCustomVerificationEmailTemplatesRequest";
const _LCVETRi = "ListCustomVerificationEmailTemplatesResponse";
const _LDDC = "ListDomainDeliverabilityCampaigns";
const _LDDCR = "ListDomainDeliverabilityCampaignsRequest";
const _LDDCRi = "ListDomainDeliverabilityCampaignsResponse";
const _LDE = "LastDeliveryEvent";
const _LDIP = "ListDedicatedIpPools";
const _LDIPR = "ListDedicatedIpPoolsRequest";
const _LDIPRi = "ListDedicatedIpPoolsResponse";
const _LDTR = "ListDeliverabilityTestReports";
const _LDTRR = "ListDeliverabilityTestReportsRequest";
const _LDTRRi = "ListDeliverabilityTestReportsResponse";
const _LEE = "LimitExceededException";
const _LEEa = "LastEngagementEvent";
const _LEI = "ListEmailIdentities";
const _LEIR = "ListEmailIdentitiesRequest";
const _LEIRi = "ListEmailIdentitiesResponse";
const _LEJ = "ListExportJobs";
const _LEJR = "ListExportJobsRequest";
const _LEJRi = "ListExportJobsResponse";
const _LET = "ListEmailTemplates";
const _LETR = "ListEmailTemplatesRequest";
const _LETRi = "ListEmailTemplatesResponse";
const _LFS = "LastFreshStart";
const _LIJ = "ListImportJobs";
const _LIJR = "ListImportJobsRequest";
const _LIJRi = "ListImportJobsResponse";
const _LKGT = "LastKeyGenerationTimestamp";
const _LMO = "ListManagementOptions";
const _LMRE = "ListMultiRegionEndpoints";
const _LMRER = "ListMultiRegionEndpointsRequest";
const _LMRERi = "ListMultiRegionEndpointsResponse";
const _LOC = "ListOfContacts";
const _LOCL = "ListOfContactLists";
const _LR = "ListRecommendations";
const _LRE = "ListReputationEntities";
const _LRER = "ListReputationEntitiesRequest";
const _LRERi = "ListReputationEntitiesResponse";
const _LRR = "ListRecommendationsRequest";
const _LRRi = "ListRecommendationsResponse";
const _LRT = "ListResourceTenants";
const _LRTR = "ListResourceTenantsRequest";
const _LRTRi = "ListResourceTenantsResponse";
const _LSD = "ListSuppressedDestinations";
const _LSDR = "ListSuppressedDestinationsRequest";
const _LSDRi = "ListSuppressedDestinationsResponse";
const _LSDT = "LastSeenDateTime";
const _LST = "LastSuccessTimestamp";
const _LT = "ListingTime";
const _LTFR = "ListTagsForResource";
const _LTFRR = "ListTagsForResourceRequest";
const _LTFRRi = "ListTagsForResourceResponse";
const _LTR = "ListTenantsRequest";
const _LTRR = "ListTenantResourcesRequest";
const _LTRRi = "ListTenantResourcesResponse";
const _LTRi = "ListTenantsResponse";
const _LTRis = "ListTenantResources";
const _LTi = "ListTenants";
const _LUT = "LastUpdatedTimestamp";
const _LUTa = "LastUpdateTime";
const _M = "Metric";
const _MDE = "MetricDataError";
const _MDEL = "MetricDataErrorList";
const _MDR = "MetricDataResult";
const _MDRL = "MetricDataResultList";
const _MDS = "MaxDeliverySeconds";
const _MDSe = "MetricsDataSource";
const _ME = "MailboxExists";
const _MET = "MatchingEventTypes";
const _MFA = "MailFromAttributes";
const _MFD = "MailFromDomain";
const _MFDNVE = "MailFromDomainNotVerifiedException";
const _MFDS = "MailFromDomainStatus";
const _MH = "MessageHeader";
const _MHL = "MessageHeaderList";
const _MHS = "Max24HourSend";
const _MI = "MessageId";
const _MIDS = "MessageInsightsDataSource";
const _MIF = "MessageInsightsFilters";
const _MP = "MissingPercentage";
const _MR = "MessageRejected";
const _MRE = "MultiRegionEndpoints";
const _MREu = "MultiRegionEndpoint";
const _MRa = "MaxResults";
const _MSR = "MaxSendRate";
const _MT = "MailType";
const _MTL = "MessageTagList";
const _MTe = "MessageTag";
const _MV = "MailboxValidation";
const _Me = "Message";
const _Met = "Metrics";
const _N = "Namespace";
const _NFE = "NotFoundException";
const _NSKL = "NextSigningKeyLength";
const _NT = "NextToken";
const _Na = "Name";
const _OCT = "OverallConfidenceThreshold";
const _OP = "OverallPlacement";
const _OSD = "OptimizedSharedDelivery";
const _OV = "OverallVolume";
const _P = "Policy";
const _PAD = "PutAccountDetails";
const _PADIWA = "PutAccountDedicatedIpWarmupAttributes";
const _PADIWAR = "PutAccountDedicatedIpWarmupAttributesRequest";
const _PADIWARu = "PutAccountDedicatedIpWarmupAttributesResponse";
const _PADR = "PutAccountDetailsRequest";
const _PADRu = "PutAccountDetailsResponse";
const _PAE = "ProductionAccessEnabled";
const _PASA = "PutAccountSendingAttributes";
const _PASAR = "PutAccountSendingAttributesRequest";
const _PASARu = "PutAccountSendingAttributesResponse";
const _PASARut = "PutAccountSuppressionAttributesRequest";
const _PASARutc = "PutAccountSuppressionAttributesResponse";
const _PASAu = "PutAccountSuppressionAttributes";
const _PAVA = "PutAccountVdmAttributes";
const _PAVAR = "PutAccountVdmAttributesRequest";
const _PAVARu = "PutAccountVdmAttributesResponse";
const _PCSAO = "PutConfigurationSetArchivingOptions";
const _PCSAOR = "PutConfigurationSetArchivingOptionsRequest";
const _PCSAORu = "PutConfigurationSetArchivingOptionsResponse";
const _PCSDO = "PutConfigurationSetDeliveryOptions";
const _PCSDOR = "PutConfigurationSetDeliveryOptionsRequest";
const _PCSDORu = "PutConfigurationSetDeliveryOptionsResponse";
const _PCSRO = "PutConfigurationSetReputationOptions";
const _PCSROR = "PutConfigurationSetReputationOptionsRequest";
const _PCSRORu = "PutConfigurationSetReputationOptionsResponse";
const _PCSSO = "PutConfigurationSetSendingOptions";
const _PCSSOR = "PutConfigurationSetSendingOptionsRequest";
const _PCSSORu = "PutConfigurationSetSendingOptionsResponse";
const _PCSSORut = "PutConfigurationSetSuppressionOptionsRequest";
const _PCSSORuto = "PutConfigurationSetSuppressionOptionsResponse";
const _PCSSOu = "PutConfigurationSetSuppressionOptions";
const _PCSTO = "PutConfigurationSetTrackingOptions";
const _PCSTOR = "PutConfigurationSetTrackingOptionsRequest";
const _PCSTORu = "PutConfigurationSetTrackingOptionsResponse";
const _PCSVO = "PutConfigurationSetVdmOptions";
const _PCSVOR = "PutConfigurationSetVdmOptionsRequest";
const _PCSVORu = "PutConfigurationSetVdmOptionsResponse";
const _PD = "PinpointDestination";
const _PDDO = "PutDeliverabilityDashboardOption";
const _PDDOR = "PutDeliverabilityDashboardOptionRequest";
const _PDDORu = "PutDeliverabilityDashboardOptionResponse";
const _PDIIP = "PutDedicatedIpInPool";
const _PDIIPR = "PutDedicatedIpInPoolRequest";
const _PDIIPRu = "PutDedicatedIpInPoolResponse";
const _PDIPSA = "PutDedicatedIpPoolScalingAttributes";
const _PDIPSAR = "PutDedicatedIpPoolScalingAttributesRequest";
const _PDIPSARu = "PutDedicatedIpPoolScalingAttributesResponse";
const _PDIWA = "PutDedicatedIpWarmupAttributes";
const _PDIWAR = "PutDedicatedIpWarmupAttributesRequest";
const _PDIWARu = "PutDedicatedIpWarmupAttributesResponse";
const _PEICSA = "PutEmailIdentityConfigurationSetAttributes";
const _PEICSAR = "PutEmailIdentityConfigurationSetAttributesRequest";
const _PEICSARu = "PutEmailIdentityConfigurationSetAttributesResponse";
const _PEIDA = "PutEmailIdentityDkimAttributes";
const _PEIDAR = "PutEmailIdentityDkimAttributesRequest";
const _PEIDARu = "PutEmailIdentityDkimAttributesResponse";
const _PEIDSA = "PutEmailIdentityDkimSigningAttributes";
const _PEIDSAR = "PutEmailIdentityDkimSigningAttributesRequest";
const _PEIDSARu = "PutEmailIdentityDkimSigningAttributesResponse";
const _PEIFA = "PutEmailIdentityFeedbackAttributes";
const _PEIFAR = "PutEmailIdentityFeedbackAttributesRequest";
const _PEIFARu = "PutEmailIdentityFeedbackAttributesResponse";
const _PEIMFA = "PutEmailIdentityMailFromAttributes";
const _PEIMFAR = "PutEmailIdentityMailFromAttributesRequest";
const _PEIMFARu = "PutEmailIdentityMailFromAttributesResponse";
const _PESD = "PendingExpirationSubscribedDomains";
const _PI = "ProjectedInbox";
const _PK = "PrivateKey";
const _PN = "PoolName";
const _PNS = "PrimaryNameServer";
const _PNo = "PolicyName";
const _PRC = "ProcessedRecordsCount";
const _PS = "PageSize";
const _PSD = "PutSuppressedDestination";
const _PSDR = "PutSuppressedDestinationRequest";
const _PSDRu = "PutSuppressedDestinationResponse";
const _PSl = "PlacementStatistics";
const _PSr = "ProjectedSpam";
const _PTSA = "PutTenantSuppressionAttributes";
const _PTSAR = "PutTenantSuppressionAttributesRequest";
const _PTSARu = "PutTenantSuppressionAttributesResponse";
const _PV = "ProjectedVolume";
const _Po = "Policies";
const _Q = "Queries";
const _R = "Results";
const _RA = "ResourceArn";
const _RC = "RawContent";
const _RD = "ReviewDetails";
const _RDR = "ReadDeleteRate";
const _RDo = "RoutesDetails";
const _RDou = "RouteDetails";
const _RE = "ReputationEntity";
const _REC = "ReplacementEmailContent";
const _REL = "ReputationEntitiesList";
const _REP = "ReputationEntityPolicy";
const _RER = "ReputationEntityReference";
const _RET = "ReputationEntityType";
const _REe = "ReputationEntities";
const _RH = "ReplacementHeaders";
const _RI = "ReportId";
const _RIe = "ReputationImpact";
const _RL = "RecommendationsList";
const _RM = "RawMessage";
const _RME = "ReputationMetricsEnabled";
const _RMP = "ReputationManagementPolicy";
const _RN = "RblName";
const _RNe = "ReportName";
const _RO = "ReputationOptions";
const _RR = "ReadRate";
const _RRP = "ReadRatePercent";
const _RT = "ReplacementTags";
const _RTA = "ReplyToAddresses";
const _RTD = "ReplacementTemplateData";
const _RTM = "ResourceTenantMetadata";
const _RTML = "ResourceTenantMetadataList";
const _RTe = "ResourceTenants";
const _RTen = "RenderedTemplate";
const _RTep = "ReplacementTemplate";
const _RTes = "ResourceType";
const _Ra = "Raw";
const _Re = "Recommendations";
const _Rea = "Reasons";
const _Reas = "Reason";
const _Rec = "Recommendation";
const _Reg = "Regions";
const _Regi = "Region";
const _Ro = "Routes";
const _Rou = "Route";
const _S = "Status";
const _SA = "SuppressionAttributes";
const _SAO = "SigningAttributesOrigin";
const _SAi = "SigningAttributes";
const _SBE = "SendBulkEmail";
const _SBER = "SendBulkEmailRequest";
const _SBERe = "SendBulkEmailResponse";
const _SC = "SpamCount";
const _SCT = "SuppressionConditionThreshold";
const _SCTu = "SuppressionConfidenceThreshold";
const _SCVE = "SendCustomVerificationEmail";
const _SCVER = "SendCustomVerificationEmailRequest";
const _SCVERe = "SendCustomVerificationEmailResponse";
const _SD = "StartDate";
const _SDA = "SuppressedDestinationAttributes";
const _SDS = "SuppressedDestinationSummaries";
const _SDSu = "SuppressedDestinationSummary";
const _SDn = "SnsDestination";
const _SDu = "SuppressedDestination";
const _SDub = "SubscribedDomain";
const _SDubs = "SubscribedDomains";
const _SE = "SigningEnabled";
const _SED = "SubscriptionExpiryDate";
const _SER = "SendEmailRequest";
const _SERe = "SendEmailResponse";
const _SEe = "SendingEnabled";
const _SEen = "SendEmail";
const _SHZ = "SigningHostedZone";
const _SI = "SendingIps";
const _SLD = "SuppressionListDestination";
const _SLH = "SentLast24Hours";
const _SLIA = "SuppressionListImportAction";
const _SM = "ScalingMode";
const _SN = "SerialNumber";
const _SO = "SendingOptions";
const _SOAR = "SOARecord";
const _SOu = "SuppressionOptions";
const _SP = "SpamPercentage";
const _SPE = "SendingPausedException";
const _SPN = "SendingPoolName";
const _SPp = "SpfPercentage";
const _SQ = "SendQuota";
const _SR = "SuppressedReasons";
const _SRC = "SpamRawCount";
const _SRURL = "SuccessRedirectionURL";
const _SRt = "StatusRecord";
const _SS = "SendingStatus";
const _SSA = "SendingStatusAggregate";
const _SSD = "SubscriptionStartDate";
const _SSu = "SuppressionScope";
const _SSub = "SubscriptionStatus";
const _SU = "S3Url";
const _SVA = "SuppressionValidationAttributes";
const _SVO = "SuppressionValidationOptions";
const _Si = "Simple";
const _St = "Statistics";
const _Su = "Subject";
const _T = "Text";
const _TA = "TenantArn";
const _TAe = "TemplateArn";
const _TAo = "ToAddresses";
const _TAop = "TopicArn";
const _TC = "TemplateContent";
const _TD = "TemplateData";
const _TDP = "TopicDefaultPreferences";
const _TF = "TopicFilter";
const _TI = "TenantId";
const _TIL = "TenantInfoList";
const _TIe = "TenantInfo";
const _TIr = "TrackedIsps";
const _TK = "TagKeys";
const _TL = "TagList";
const _TM = "TemplatesMetadata";
const _TMRE = "TooManyRequestsException";
const _TN = "TemplateName";
const _TNe = "TenantName";
const _TNo = "TopicName";
const _TO = "TrackingOptions";
const _TP = "TopicPreferences";
const _TPL = "TopicPreferenceList";
const _TPl = "TlsPolicy";
const _TPo = "TopicPreference";
const _TR = "TenantResources";
const _TRET = "TestRenderEmailTemplate";
const _TRETR = "TestRenderEmailTemplateRequest";
const _TRETRe = "TestRenderEmailTemplateResponse";
const _TRL = "TenantResourceList";
const _TRR = "TagResourceRequest";
const _TRRa = "TagResourceResponse";
const _TRa = "TagResource";
const _TRe = "TenantResource";
const _TS = "TemplateSubject";
const _TSA = "TenantSuppressionAttributes";
const _Ta = "Tags";
const _Tag = "Tag";
const _Te = "Template";
const _Ten = "Tenant";
const _Tena = "Tenants";
const _Ti = "Timestamp";
const _Tim = "Timestamps";
const _To = "Topics";
const _Tok = "Tokens";
const _Top = "Topic";
const _Ty = "Type";
const _UA = "UnsubscribeAll";
const _UC = "UpdateContact";
const _UCD = "UseCaseDescription";
const _UCL = "UpdateContactList";
const _UCLR = "UpdateContactListRequest";
const _UCLRp = "UpdateContactListResponse";
const _UCR = "UpdateContactRequest";
const _UCRp = "UpdateContactResponse";
const _UCSED = "UpdateConfigurationSetEventDestination";
const _UCSEDR = "UpdateConfigurationSetEventDestinationRequest";
const _UCSEDRp = "UpdateConfigurationSetEventDestinationResponse";
const _UCVET = "UpdateCustomVerificationEmailTemplate";
const _UCVETR = "UpdateCustomVerificationEmailTemplateRequest";
const _UCVETRp = "UpdateCustomVerificationEmailTemplateResponse";
const _UDIPU = "UseDefaultIfPreferenceUnavailable";
const _UEIP = "UpdateEmailIdentityPolicy";
const _UEIPR = "UpdateEmailIdentityPolicyRequest";
const _UEIPRp = "UpdateEmailIdentityPolicyResponse";
const _UET = "UpdateEmailTemplate";
const _UETR = "UpdateEmailTemplateRequest";
const _UETRp = "UpdateEmailTemplateResponse";
const _UR = "UntagResource";
const _URECMS = "UpdateReputationEntityCustomerManagedStatus";
const _URECMSR = "UpdateReputationEntityCustomerManagedStatusRequest";
const _URECMSRp = "UpdateReputationEntityCustomerManagedStatusResponse";
const _UREP = "UpdateReputationEntityPolicy";
const _UREPR = "UpdateReputationEntityPolicyRequest";
const _UREPRp = "UpdateReputationEntityPolicyResponse";
const _URR = "UntagResourceRequest";
const _URRn = "UntagResourceResponse";
const _V = "Value";
const _VA = "VdmAttributes";
const _VAa = "ValidationAttributes";
const _VE = "VdmEnabled";
const _VFSS = "VerifiedForSendingStatus";
const _VI = "VerificationInfo";
const _VO = "VdmOptions";
const _VOa = "ValidationOptions";
const _VS = "VolumeStatistics";
const _VSe = "VerificationStatus";
const _Va = "Values";
const _WP = "WarmupPercentage";
const _WS = "WarmupStatus";
const _WURL = "WebsiteURL";
const _c = "client";
const _e = "error";
const _h = "http";
const _hE = "httpError";
const _hQ = "httpQuery";
const _m = "message";
const _s = "smithy.ts.sdk.synthetic.com.amazonaws.sesv2";
const _se = "server";
const n0 = "com.amazonaws.sesv2";
const { TypeRegistry } = require("@smithy/core/schema");
const { AccountSuspendedException, AlreadyExistsException, BadRequestException, ConcurrentModificationException, ConflictException, InternalServiceErrorException, InvalidNextTokenException, LimitExceededException, MailFromDomainNotVerifiedException, MessageRejected, NotFoundException, SendingPausedException, TooManyRequestsException } = require("../models/errors");
const { SESv2ServiceException } = require("../models/SESv2ServiceException");
const _s_registry = TypeRegistry.for(_s);
const SESv2ServiceException$ = [-3, _s, "SESv2ServiceException", 0, [], []];
exports.SESv2ServiceException$ = SESv2ServiceException$;
_s_registry.registerError(SESv2ServiceException$, SESv2ServiceException);
const n0_registry = TypeRegistry.for(n0);
const AccountSuspendedException$ = [-3, n0, _ASE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.AccountSuspendedException$ = AccountSuspendedException$;
n0_registry.registerError(AccountSuspendedException$, AccountSuspendedException);
const AlreadyExistsException$ = [-3, n0, _AEE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.AlreadyExistsException$ = AlreadyExistsException$;
n0_registry.registerError(AlreadyExistsException$, AlreadyExistsException);
const BadRequestException$ = [-3, n0, _BRE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.BadRequestException$ = BadRequestException$;
n0_registry.registerError(BadRequestException$, BadRequestException);
const ConcurrentModificationException$ = [-3, n0, _CME,
    { [_e]: _se, [_hE]: 500 },
    [_m],
    [0]
];
exports.ConcurrentModificationException$ = ConcurrentModificationException$;
n0_registry.registerError(ConcurrentModificationException$, ConcurrentModificationException);
const ConflictException$ = [-3, n0, _CE,
    { [_e]: _c, [_hE]: 409 },
    [_m],
    [0]
];
exports.ConflictException$ = ConflictException$;
n0_registry.registerError(ConflictException$, ConflictException);
const InternalServiceErrorException$ = [-3, n0, _ISEE,
    { [_e]: _se, [_hE]: 500 },
    [_m],
    [0]
];
exports.InternalServiceErrorException$ = InternalServiceErrorException$;
n0_registry.registerError(InternalServiceErrorException$, InternalServiceErrorException);
const InvalidNextTokenException$ = [-3, n0, _INTE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.InvalidNextTokenException$ = InvalidNextTokenException$;
n0_registry.registerError(InvalidNextTokenException$, InvalidNextTokenException);
const LimitExceededException$ = [-3, n0, _LEE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.LimitExceededException$ = LimitExceededException$;
n0_registry.registerError(LimitExceededException$, LimitExceededException);
const MailFromDomainNotVerifiedException$ = [-3, n0, _MFDNVE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.MailFromDomainNotVerifiedException$ = MailFromDomainNotVerifiedException$;
n0_registry.registerError(MailFromDomainNotVerifiedException$, MailFromDomainNotVerifiedException);
const MessageRejected$ = [-3, n0, _MR,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.MessageRejected$ = MessageRejected$;
n0_registry.registerError(MessageRejected$, MessageRejected);
const NotFoundException$ = [-3, n0, _NFE,
    { [_e]: _c, [_hE]: 404 },
    [_m],
    [0]
];
exports.NotFoundException$ = NotFoundException$;
n0_registry.registerError(NotFoundException$, NotFoundException);
const SendingPausedException$ = [-3, n0, _SPE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
exports.SendingPausedException$ = SendingPausedException$;
n0_registry.registerError(SendingPausedException$, SendingPausedException);
const TooManyRequestsException$ = [-3, n0, _TMRE,
    { [_e]: _c, [_hE]: 429 },
    [_m],
    [0]
];
exports.TooManyRequestsException$ = TooManyRequestsException$;
n0_registry.registerError(TooManyRequestsException$, TooManyRequestsException);
exports.errorTypeRegistries = [
    _s_registry,
    n0_registry,
];
var AdditionalContactEmailAddress = [0, n0, _ACEA, 8, 0];
var EmailSubject = [0, n0, _ES, 8, 0];
var InsightsEmailAddress = [0, n0, _IEA, 8, 0];
var PrivateKey = [0, n0, _PK, 8, 0];
var UseCaseDescription = [0, n0, _UCD, 8, 0];
var WebsiteURL = [0, n0, _WURL, 8, 0];
const AccountDetails$ = [3, n0, _AD,
    0,
    [_MT, _WURL, _CL, _UCD, _ACEAd, _RD],
    [0, [() => WebsiteURL, 0], 0, [() => UseCaseDescription, 0], [() => AdditionalContactEmailAddresses, 0], () => ReviewDetails$]
];
exports.AccountDetails$ = AccountDetails$;
const ArchivingOptions$ = [3, n0, _AO,
    0,
    [_AA],
    [0]
];
exports.ArchivingOptions$ = ArchivingOptions$;
const Attachment$ = [3, n0, _A,
    0,
    [_RC, _FN, _CD, _CDo, _CI, _CTE, _CT],
    [21, 0, 0, 0, 0, 0, 0], 2
];
exports.Attachment$ = Attachment$;
const BatchGetMetricDataQuery$ = [3, n0, _BGMDQ,
    0,
    [_I, _N, _M, _SD, _ED, _D],
    [0, 0, 0, 4, 4, 128 | 0], 5
];
exports.BatchGetMetricDataQuery$ = BatchGetMetricDataQuery$;
const BatchGetMetricDataRequest$ = [3, n0, _BGMDR,
    0,
    [_Q],
    [() => BatchGetMetricDataQueries], 1
];
exports.BatchGetMetricDataRequest$ = BatchGetMetricDataRequest$;
const BatchGetMetricDataResponse$ = [3, n0, _BGMDRa,
    0,
    [_R, _E],
    [() => MetricDataResultList, () => MetricDataErrorList]
];
exports.BatchGetMetricDataResponse$ = BatchGetMetricDataResponse$;
const BlacklistEntry$ = [3, n0, _BE,
    0,
    [_RN, _LT, _De],
    [0, 4, 0]
];
exports.BlacklistEntry$ = BlacklistEntry$;
const Body$ = [3, n0, _B,
    0,
    [_T, _H],
    [() => Content$, () => Content$]
];
exports.Body$ = Body$;
const Bounce$ = [3, n0, _Bo,
    0,
    [_BT, _BST, _DC],
    [0, 0, 0]
];
exports.Bounce$ = Bounce$;
const BulkEmailContent$ = [3, n0, _BEC,
    0,
    [_Te],
    [() => Template$]
];
exports.BulkEmailContent$ = BulkEmailContent$;
const BulkEmailEntry$ = [3, n0, _BEE,
    0,
    [_Des, _RT, _REC, _RH],
    [() => Destination$, () => MessageTagList, () => ReplacementEmailContent$, () => MessageHeaderList], 1
];
exports.BulkEmailEntry$ = BulkEmailEntry$;
const BulkEmailEntryResult$ = [3, n0, _BEER,
    0,
    [_S, _Er, _MI],
    [0, 0, 0]
];
exports.BulkEmailEntryResult$ = BulkEmailEntryResult$;
const CancelExportJobRequest$ = [3, n0, _CEJR,
    0,
    [_JI],
    [[0, 1]], 1
];
exports.CancelExportJobRequest$ = CancelExportJobRequest$;
const CancelExportJobResponse$ = [3, n0, _CEJRa,
    0,
    [],
    []
];
exports.CancelExportJobResponse$ = CancelExportJobResponse$;
const CloudWatchDestination$ = [3, n0, _CWD,
    0,
    [_DCi],
    [() => CloudWatchDimensionConfigurations], 1
];
exports.CloudWatchDestination$ = CloudWatchDestination$;
const CloudWatchDimensionConfiguration$ = [3, n0, _CWDC,
    0,
    [_DN, _DVS, _DDV],
    [0, 0, 0], 3
];
exports.CloudWatchDimensionConfiguration$ = CloudWatchDimensionConfiguration$;
const Complaint$ = [3, n0, _C,
    0,
    [_CST, _CFT],
    [0, 0]
];
exports.Complaint$ = Complaint$;
const Contact$ = [3, n0, _Co,
    0,
    [_EA, _TP, _TDP, _UA, _LUT],
    [0, () => TopicPreferenceList, () => TopicPreferenceList, 2, 4]
];
exports.Contact$ = Contact$;
const ContactList$ = [3, n0, _CLo,
    0,
    [_CLN, _LUT],
    [0, 4]
];
exports.ContactList$ = ContactList$;
const ContactListDestination$ = [3, n0, _CLD,
    0,
    [_CLN, _CLIA],
    [0, 0], 2
];
exports.ContactListDestination$ = ContactListDestination$;
const Content$ = [3, n0, _Con,
    0,
    [_Da, _Ch],
    [0, 0], 1
];
exports.Content$ = Content$;
const CreateConfigurationSetEventDestinationRequest$ = [3, n0, _CCSEDR,
    0,
    [_CSN, _EDN, _EDv],
    [[0, 1], 0, () => EventDestinationDefinition$], 3
];
exports.CreateConfigurationSetEventDestinationRequest$ = CreateConfigurationSetEventDestinationRequest$;
const CreateConfigurationSetEventDestinationResponse$ = [3, n0, _CCSEDRr,
    0,
    [],
    []
];
exports.CreateConfigurationSetEventDestinationResponse$ = CreateConfigurationSetEventDestinationResponse$;
const CreateConfigurationSetRequest$ = [3, n0, _CCSR,
    0,
    [_CSN, _TO, _DO, _RO, _SO, _Ta, _SOu, _VO, _AO],
    [0, () => TrackingOptions$, () => DeliveryOptions$, () => ReputationOptions$, () => SendingOptions$, () => TagList, () => SuppressionOptions$, () => VdmOptions$, () => ArchivingOptions$], 1
];
exports.CreateConfigurationSetRequest$ = CreateConfigurationSetRequest$;
const CreateConfigurationSetResponse$ = [3, n0, _CCSRr,
    0,
    [],
    []
];
exports.CreateConfigurationSetResponse$ = CreateConfigurationSetResponse$;
const CreateContactListRequest$ = [3, n0, _CCLR,
    0,
    [_CLN, _To, _De, _Ta],
    [0, () => Topics, 0, () => TagList], 1
];
exports.CreateContactListRequest$ = CreateContactListRequest$;
const CreateContactListResponse$ = [3, n0, _CCLRr,
    0,
    [],
    []
];
exports.CreateContactListResponse$ = CreateContactListResponse$;
const CreateContactRequest$ = [3, n0, _CCR,
    0,
    [_CLN, _EA, _TP, _UA, _ADt],
    [[0, 1], 0, () => TopicPreferenceList, 2, 0], 2
];
exports.CreateContactRequest$ = CreateContactRequest$;
const CreateContactResponse$ = [3, n0, _CCRr,
    0,
    [],
    []
];
exports.CreateContactResponse$ = CreateContactResponse$;
const CreateCustomVerificationEmailTemplateRequest$ = [3, n0, _CCVETR,
    0,
    [_TN, _FEA, _TS, _TC, _SRURL, _FRURL, _Ta],
    [0, 0, 0, 0, 0, 0, () => TagList], 6
];
exports.CreateCustomVerificationEmailTemplateRequest$ = CreateCustomVerificationEmailTemplateRequest$;
const CreateCustomVerificationEmailTemplateResponse$ = [3, n0, _CCVETRr,
    0,
    [],
    []
];
exports.CreateCustomVerificationEmailTemplateResponse$ = CreateCustomVerificationEmailTemplateResponse$;
const CreateDedicatedIpPoolRequest$ = [3, n0, _CDIPR,
    0,
    [_PN, _Ta, _SM],
    [0, () => TagList, 0], 1
];
exports.CreateDedicatedIpPoolRequest$ = CreateDedicatedIpPoolRequest$;
const CreateDedicatedIpPoolResponse$ = [3, n0, _CDIPRr,
    0,
    [],
    []
];
exports.CreateDedicatedIpPoolResponse$ = CreateDedicatedIpPoolResponse$;
const CreateDeliverabilityTestReportRequest$ = [3, n0, _CDTRR,
    0,
    [_FEA, _Con, _RNe, _Ta],
    [0, () => EmailContent$, 0, () => TagList], 2
];
exports.CreateDeliverabilityTestReportRequest$ = CreateDeliverabilityTestReportRequest$;
const CreateDeliverabilityTestReportResponse$ = [3, n0, _CDTRRr,
    0,
    [_RI, _DTS],
    [0, 0], 2
];
exports.CreateDeliverabilityTestReportResponse$ = CreateDeliverabilityTestReportResponse$;
const CreateEmailIdentityPolicyRequest$ = [3, n0, _CEIPR,
    0,
    [_EI, _PNo, _P],
    [[0, 1], [0, 1], 0], 3
];
exports.CreateEmailIdentityPolicyRequest$ = CreateEmailIdentityPolicyRequest$;
const CreateEmailIdentityPolicyResponse$ = [3, n0, _CEIPRr,
    0,
    [],
    []
];
exports.CreateEmailIdentityPolicyResponse$ = CreateEmailIdentityPolicyResponse$;
const CreateEmailIdentityRequest$ = [3, n0, _CEIR,
    0,
    [_EI, _Ta, _DSA, _CSN],
    [0, () => TagList, [() => DkimSigningAttributes$, 0], 0], 1
];
exports.CreateEmailIdentityRequest$ = CreateEmailIdentityRequest$;
const CreateEmailIdentityResponse$ = [3, n0, _CEIRr,
    0,
    [_IT, _VFSS, _DA],
    [0, 2, () => DkimAttributes$]
];
exports.CreateEmailIdentityResponse$ = CreateEmailIdentityResponse$;
const CreateEmailTemplateRequest$ = [3, n0, _CETR,
    0,
    [_TN, _TC, _Ta],
    [0, () => EmailTemplateContent$, () => TagList], 2
];
exports.CreateEmailTemplateRequest$ = CreateEmailTemplateRequest$;
const CreateEmailTemplateResponse$ = [3, n0, _CETRr,
    0,
    [],
    []
];
exports.CreateEmailTemplateResponse$ = CreateEmailTemplateResponse$;
const CreateExportJobRequest$ = [3, n0, _CEJRr,
    0,
    [_EDS, _EDx],
    [[() => ExportDataSource$, 0], () => ExportDestination$], 2
];
exports.CreateExportJobRequest$ = CreateExportJobRequest$;
const CreateExportJobResponse$ = [3, n0, _CEJRre,
    0,
    [_JI],
    [0]
];
exports.CreateExportJobResponse$ = CreateExportJobResponse$;
const CreateImportJobRequest$ = [3, n0, _CIJR,
    0,
    [_ID, _IDS],
    [() => ImportDestination$, () => ImportDataSource$], 2
];
exports.CreateImportJobRequest$ = CreateImportJobRequest$;
const CreateImportJobResponse$ = [3, n0, _CIJRr,
    0,
    [_JI],
    [0]
];
exports.CreateImportJobResponse$ = CreateImportJobResponse$;
const CreateMultiRegionEndpointRequest$ = [3, n0, _CMRER,
    0,
    [_EN, _Det, _Ta],
    [0, () => Details$, () => TagList], 2
];
exports.CreateMultiRegionEndpointRequest$ = CreateMultiRegionEndpointRequest$;
const CreateMultiRegionEndpointResponse$ = [3, n0, _CMRERr,
    0,
    [_S, _EIn],
    [0, 0]
];
exports.CreateMultiRegionEndpointResponse$ = CreateMultiRegionEndpointResponse$;
const CreateTenantRequest$ = [3, n0, _CTR,
    0,
    [_TNe, _Ta, _SA],
    [0, () => TagList, () => TenantSuppressionAttributes$], 1
];
exports.CreateTenantRequest$ = CreateTenantRequest$;
const CreateTenantResourceAssociationRequest$ = [3, n0, _CTRAR,
    0,
    [_TNe, _RA],
    [0, 0], 2
];
exports.CreateTenantResourceAssociationRequest$ = CreateTenantResourceAssociationRequest$;
const CreateTenantResourceAssociationResponse$ = [3, n0, _CTRARr,
    0,
    [],
    []
];
exports.CreateTenantResourceAssociationResponse$ = CreateTenantResourceAssociationResponse$;
const CreateTenantResponse$ = [3, n0, _CTRr,
    0,
    [_TNe, _TI, _TA, _CTr, _Ta, _SS, _SA],
    [0, 0, 0, 4, () => TagList, 0, () => TenantSuppressionAttributes$]
];
exports.CreateTenantResponse$ = CreateTenantResponse$;
const CustomVerificationEmailTemplateMetadata$ = [3, n0, _CVETM,
    0,
    [_TN, _FEA, _TS, _SRURL, _FRURL],
    [0, 0, 0, 0, 0]
];
exports.CustomVerificationEmailTemplateMetadata$ = CustomVerificationEmailTemplateMetadata$;
const DailyVolume$ = [3, n0, _DV,
    0,
    [_SD, _VS, _DIP],
    [4, () => VolumeStatistics$, () => DomainIspPlacements]
];
exports.DailyVolume$ = DailyVolume$;
const DashboardAttributes$ = [3, n0, _DAa,
    0,
    [_EM],
    [0]
];
exports.DashboardAttributes$ = DashboardAttributes$;
const DashboardOptions$ = [3, n0, _DOa,
    0,
    [_EM],
    [0]
];
exports.DashboardOptions$ = DashboardOptions$;
const DedicatedIp$ = [3, n0, _DI,
    0,
    [_Ip, _WS, _WP, _PN],
    [0, 0, 1, 0], 3
];
exports.DedicatedIp$ = DedicatedIp$;
const DedicatedIpPool$ = [3, n0, _DIPe,
    0,
    [_PN, _SM],
    [0, 0], 2
];
exports.DedicatedIpPool$ = DedicatedIpPool$;
const DeleteConfigurationSetEventDestinationRequest$ = [3, n0, _DCSEDR,
    0,
    [_CSN, _EDN],
    [[0, 1], [0, 1]], 2
];
exports.DeleteConfigurationSetEventDestinationRequest$ = DeleteConfigurationSetEventDestinationRequest$;
const DeleteConfigurationSetEventDestinationResponse$ = [3, n0, _DCSEDRe,
    0,
    [],
    []
];
exports.DeleteConfigurationSetEventDestinationResponse$ = DeleteConfigurationSetEventDestinationResponse$;
const DeleteConfigurationSetRequest$ = [3, n0, _DCSR,
    0,
    [_CSN],
    [[0, 1]], 1
];
exports.DeleteConfigurationSetRequest$ = DeleteConfigurationSetRequest$;
const DeleteConfigurationSetResponse$ = [3, n0, _DCSRe,
    0,
    [],
    []
];
exports.DeleteConfigurationSetResponse$ = DeleteConfigurationSetResponse$;
const DeleteContactListRequest$ = [3, n0, _DCLR,
    0,
    [_CLN],
    [[0, 1]], 1
];
exports.DeleteContactListRequest$ = DeleteContactListRequest$;
const DeleteContactListResponse$ = [3, n0, _DCLRe,
    0,
    [],
    []
];
exports.DeleteContactListResponse$ = DeleteContactListResponse$;
const DeleteContactRequest$ = [3, n0, _DCR,
    0,
    [_CLN, _EA],
    [[0, 1], [0, 1]], 2
];
exports.DeleteContactRequest$ = DeleteContactRequest$;
const DeleteContactResponse$ = [3, n0, _DCRe,
    0,
    [],
    []
];
exports.DeleteContactResponse$ = DeleteContactResponse$;
const DeleteCustomVerificationEmailTemplateRequest$ = [3, n0, _DCVETR,
    0,
    [_TN],
    [[0, 1]], 1
];
exports.DeleteCustomVerificationEmailTemplateRequest$ = DeleteCustomVerificationEmailTemplateRequest$;
const DeleteCustomVerificationEmailTemplateResponse$ = [3, n0, _DCVETRe,
    0,
    [],
    []
];
exports.DeleteCustomVerificationEmailTemplateResponse$ = DeleteCustomVerificationEmailTemplateResponse$;
const DeleteDedicatedIpPoolRequest$ = [3, n0, _DDIPR,
    0,
    [_PN],
    [[0, 1]], 1
];
exports.DeleteDedicatedIpPoolRequest$ = DeleteDedicatedIpPoolRequest$;
const DeleteDedicatedIpPoolResponse$ = [3, n0, _DDIPRe,
    0,
    [],
    []
];
exports.DeleteDedicatedIpPoolResponse$ = DeleteDedicatedIpPoolResponse$;
const DeleteEmailIdentityPolicyRequest$ = [3, n0, _DEIPR,
    0,
    [_EI, _PNo],
    [[0, 1], [0, 1]], 2
];
exports.DeleteEmailIdentityPolicyRequest$ = DeleteEmailIdentityPolicyRequest$;
const DeleteEmailIdentityPolicyResponse$ = [3, n0, _DEIPRe,
    0,
    [],
    []
];
exports.DeleteEmailIdentityPolicyResponse$ = DeleteEmailIdentityPolicyResponse$;
const DeleteEmailIdentityRequest$ = [3, n0, _DEIR,
    0,
    [_EI],
    [[0, 1]], 1
];
exports.DeleteEmailIdentityRequest$ = DeleteEmailIdentityRequest$;
const DeleteEmailIdentityResponse$ = [3, n0, _DEIRe,
    0,
    [],
    []
];
exports.DeleteEmailIdentityResponse$ = DeleteEmailIdentityResponse$;
const DeleteEmailTemplateRequest$ = [3, n0, _DETR,
    0,
    [_TN],
    [[0, 1]], 1
];
exports.DeleteEmailTemplateRequest$ = DeleteEmailTemplateRequest$;
const DeleteEmailTemplateResponse$ = [3, n0, _DETRe,
    0,
    [],
    []
];
exports.DeleteEmailTemplateResponse$ = DeleteEmailTemplateResponse$;
const DeleteMultiRegionEndpointRequest$ = [3, n0, _DMRER,
    0,
    [_EN],
    [[0, 1]], 1
];
exports.DeleteMultiRegionEndpointRequest$ = DeleteMultiRegionEndpointRequest$;
const DeleteMultiRegionEndpointResponse$ = [3, n0, _DMRERe,
    0,
    [_S],
    [0]
];
exports.DeleteMultiRegionEndpointResponse$ = DeleteMultiRegionEndpointResponse$;
const DeleteSuppressedDestinationRequest$ = [3, n0, _DSDR,
    0,
    [_EA, _TNe],
    [[0, 1], [0, { [_hQ]: _TNe }]], 1
];
exports.DeleteSuppressedDestinationRequest$ = DeleteSuppressedDestinationRequest$;
const DeleteSuppressedDestinationResponse$ = [3, n0, _DSDRe,
    0,
    [],
    []
];
exports.DeleteSuppressedDestinationResponse$ = DeleteSuppressedDestinationResponse$;
const DeleteTenantRequest$ = [3, n0, _DTR,
    0,
    [_TNe],
    [0], 1
];
exports.DeleteTenantRequest$ = DeleteTenantRequest$;
const DeleteTenantResourceAssociationRequest$ = [3, n0, _DTRAR,
    0,
    [_TNe, _RA],
    [0, 0], 2
];
exports.DeleteTenantResourceAssociationRequest$ = DeleteTenantResourceAssociationRequest$;
const DeleteTenantResourceAssociationResponse$ = [3, n0, _DTRARe,
    0,
    [],
    []
];
exports.DeleteTenantResourceAssociationResponse$ = DeleteTenantResourceAssociationResponse$;
const DeleteTenantResponse$ = [3, n0, _DTRe,
    0,
    [],
    []
];
exports.DeleteTenantResponse$ = DeleteTenantResponse$;
const DeliverabilityTestReport$ = [3, n0, _DTRel,
    0,
    [_RI, _RNe, _Su, _FEA, _CDr, _DTS],
    [0, 0, 0, 0, 4, 0]
];
exports.DeliverabilityTestReport$ = DeliverabilityTestReport$;
const DeliveryOptions$ = [3, n0, _DO,
    0,
    [_TPl, _SPN, _MDS],
    [0, 0, 1]
];
exports.DeliveryOptions$ = DeliveryOptions$;
const Destination$ = [3, n0, _Des,
    0,
    [_TAo, _CA, _BA],
    [64 | 0, 64 | 0, 64 | 0]
];
exports.Destination$ = Destination$;
const Details$ = [3, n0, _Det,
    0,
    [_RDo],
    [() => RoutesDetails], 1
];
exports.Details$ = Details$;
const DkimAttributes$ = [3, n0, _DA,
    0,
    [_SE, _S, _Tok, _SHZ, _SAO, _NSKL, _CSKL, _LKGT],
    [2, 0, 64 | 0, 0, 0, 0, 0, 4]
];
exports.DkimAttributes$ = DkimAttributes$;
const DkimSigningAttributes$ = [3, n0, _DSA,
    0,
    [_DSS, _DSPK, _NSKL, _DSAO],
    [0, [() => PrivateKey, 0], 0, 0]
];
exports.DkimSigningAttributes$ = DkimSigningAttributes$;
const DomainDeliverabilityCampaign$ = [3, n0, _DDC,
    0,
    [_CIa, _IU, _Su, _FA, _SI, _FSDT, _LSDT, _IC, _SC, _RR, _DR, _RDR, _PV, _Es],
    [0, 0, 0, 0, 64 | 0, 4, 4, 1, 1, 1, 1, 1, 1, 64 | 0]
];
exports.DomainDeliverabilityCampaign$ = DomainDeliverabilityCampaign$;
const DomainDeliverabilityTrackingOption$ = [3, n0, _DDTO,
    0,
    [_Do, _SSD, _IPTO],
    [0, 4, () => InboxPlacementTrackingOption$]
];
exports.DomainDeliverabilityTrackingOption$ = DomainDeliverabilityTrackingOption$;
const DomainIspPlacement$ = [3, n0, _DIPo,
    0,
    [_IN, _IRC, _SRC, _IP, _SP],
    [0, 1, 1, 1, 1]
];
exports.DomainIspPlacement$ = DomainIspPlacement$;
const EmailAddressInsightsMailboxEvaluations$ = [3, n0, _EAIME,
    0,
    [_HVS, _HVDR, _ME, _IRA, _IDs, _IRI],
    [() => EmailAddressInsightsVerdict$, () => EmailAddressInsightsVerdict$, () => EmailAddressInsightsVerdict$, () => EmailAddressInsightsVerdict$, () => EmailAddressInsightsVerdict$, () => EmailAddressInsightsVerdict$]
];
exports.EmailAddressInsightsMailboxEvaluations$ = EmailAddressInsightsMailboxEvaluations$;
const EmailAddressInsightsVerdict$ = [3, n0, _EAIV,
    0,
    [_CV],
    [0]
];
exports.EmailAddressInsightsVerdict$ = EmailAddressInsightsVerdict$;
const EmailContent$ = [3, n0, _EC,
    0,
    [_Si, _Ra, _Te],
    [() => Message$, () => RawMessage$, () => Template$]
];
exports.EmailContent$ = EmailContent$;
const EmailInsights$ = [3, n0, _EIm,
    0,
    [_Des, _Is, _Ev],
    [[() => InsightsEmailAddress, 0], 0, () => InsightsEvents]
];
exports.EmailInsights$ = EmailInsights$;
const EmailTemplateContent$ = [3, n0, _ETC,
    0,
    [_Su, _T, _H],
    [0, 0, 0]
];
exports.EmailTemplateContent$ = EmailTemplateContent$;
const EmailTemplateMetadata$ = [3, n0, _ETM,
    0,
    [_TN, _CTr],
    [0, 4]
];
exports.EmailTemplateMetadata$ = EmailTemplateMetadata$;
const EventBridgeDestination$ = [3, n0, _EBD,
    0,
    [_EBA],
    [0], 1
];
exports.EventBridgeDestination$ = EventBridgeDestination$;
const EventDestination$ = [3, n0, _EDv,
    0,
    [_Na, _MET, _En, _KFD, _CWD, _SDn, _EBD, _PD],
    [0, 64 | 0, 2, () => KinesisFirehoseDestination$, () => CloudWatchDestination$, () => SnsDestination$, () => EventBridgeDestination$, () => PinpointDestination$], 2
];
exports.EventDestination$ = EventDestination$;
const EventDestinationDefinition$ = [3, n0, _EDD,
    0,
    [_En, _MET, _KFD, _CWD, _SDn, _EBD, _PD],
    [2, 64 | 0, () => KinesisFirehoseDestination$, () => CloudWatchDestination$, () => SnsDestination$, () => EventBridgeDestination$, () => PinpointDestination$]
];
exports.EventDestinationDefinition$ = EventDestinationDefinition$;
const EventDetails$ = [3, n0, _EDve,
    0,
    [_Bo, _C],
    [() => Bounce$, () => Complaint$]
];
exports.EventDetails$ = EventDetails$;
const ExportDataSource$ = [3, n0, _EDS,
    0,
    [_MDSe, _MIDS],
    [() => MetricsDataSource$, [() => MessageInsightsDataSource$, 0]]
];
exports.ExportDataSource$ = ExportDataSource$;
const ExportDestination$ = [3, n0, _EDx,
    0,
    [_DF, _SU],
    [0, 0], 1
];
exports.ExportDestination$ = ExportDestination$;
const ExportJobSummary$ = [3, n0, _EJS,
    0,
    [_JI, _EST, _JS, _CTr, _CTo],
    [0, 0, 0, 4, 4]
];
exports.ExportJobSummary$ = ExportJobSummary$;
const ExportMetric$ = [3, n0, _EMx,
    0,
    [_Na, _Ag],
    [0, 0]
];
exports.ExportMetric$ = ExportMetric$;
const ExportStatistics$ = [3, n0, _ESx,
    0,
    [_PRC, _ERC],
    [1, 1]
];
exports.ExportStatistics$ = ExportStatistics$;
const FailureInfo$ = [3, n0, _FI,
    0,
    [_FRSU, _EMr],
    [0, 0]
];
exports.FailureInfo$ = FailureInfo$;
const GetAccountRequest$ = [3, n0, _GAR,
    0,
    [],
    []
];
exports.GetAccountRequest$ = GetAccountRequest$;
const GetAccountResponse$ = [3, n0, _GARe,
    0,
    [_DIAWE, _ESn, _PAE, _SQ, _SEe, _SA, _Det, _VA],
    [2, 0, 2, () => SendQuota$, 2, () => SuppressionAttributes$, [() => AccountDetails$, 0], () => VdmAttributes$]
];
exports.GetAccountResponse$ = GetAccountResponse$;
const GetBlacklistReportsRequest$ = [3, n0, _GBRR,
    0,
    [_BIN],
    [[64 | 0, { [_hQ]: _BIN }]], 1
];
exports.GetBlacklistReportsRequest$ = GetBlacklistReportsRequest$;
const GetBlacklistReportsResponse$ = [3, n0, _GBRRe,
    0,
    [_BR],
    [() => BlacklistReport], 1
];
exports.GetBlacklistReportsResponse$ = GetBlacklistReportsResponse$;
const GetConfigurationSetEventDestinationsRequest$ = [3, n0, _GCSEDR,
    0,
    [_CSN],
    [[0, 1]], 1
];
exports.GetConfigurationSetEventDestinationsRequest$ = GetConfigurationSetEventDestinationsRequest$;
const GetConfigurationSetEventDestinationsResponse$ = [3, n0, _GCSEDRe,
    0,
    [_EDven],
    [() => EventDestinations]
];
exports.GetConfigurationSetEventDestinationsResponse$ = GetConfigurationSetEventDestinationsResponse$;
const GetConfigurationSetRequest$ = [3, n0, _GCSR,
    0,
    [_CSN],
    [[0, 1]], 1
];
exports.GetConfigurationSetRequest$ = GetConfigurationSetRequest$;
const GetConfigurationSetResponse$ = [3, n0, _GCSRe,
    0,
    [_CSN, _TO, _DO, _RO, _SO, _Ta, _SOu, _VO, _AO],
    [0, () => TrackingOptions$, () => DeliveryOptions$, () => ReputationOptions$, () => SendingOptions$, () => TagList, () => SuppressionOptions$, () => VdmOptions$, () => ArchivingOptions$]
];
exports.GetConfigurationSetResponse$ = GetConfigurationSetResponse$;
const GetContactListRequest$ = [3, n0, _GCLR,
    0,
    [_CLN],
    [[0, 1]], 1
];
exports.GetContactListRequest$ = GetContactListRequest$;
const GetContactListResponse$ = [3, n0, _GCLRe,
    0,
    [_CLN, _To, _De, _CTr, _LUT, _Ta],
    [0, () => Topics, 0, 4, 4, () => TagList]
];
exports.GetContactListResponse$ = GetContactListResponse$;
const GetContactRequest$ = [3, n0, _GCR,
    0,
    [_CLN, _EA],
    [[0, 1], [0, 1]], 2
];
exports.GetContactRequest$ = GetContactRequest$;
const GetContactResponse$ = [3, n0, _GCRe,
    0,
    [_CLN, _EA, _TP, _TDP, _UA, _ADt, _CTr, _LUT],
    [0, 0, () => TopicPreferenceList, () => TopicPreferenceList, 2, 0, 4, 4]
];
exports.GetContactResponse$ = GetContactResponse$;
const GetCustomVerificationEmailTemplateRequest$ = [3, n0, _GCVETR,
    0,
    [_TN],
    [[0, 1]], 1
];
exports.GetCustomVerificationEmailTemplateRequest$ = GetCustomVerificationEmailTemplateRequest$;
const GetCustomVerificationEmailTemplateResponse$ = [3, n0, _GCVETRe,
    0,
    [_TN, _FEA, _TS, _TC, _Ta, _SRURL, _FRURL],
    [0, 0, 0, 0, () => TagList, 0, 0]
];
exports.GetCustomVerificationEmailTemplateResponse$ = GetCustomVerificationEmailTemplateResponse$;
const GetDedicatedIpPoolRequest$ = [3, n0, _GDIPR,
    0,
    [_PN],
    [[0, 1]], 1
];
exports.GetDedicatedIpPoolRequest$ = GetDedicatedIpPoolRequest$;
const GetDedicatedIpPoolResponse$ = [3, n0, _GDIPRe,
    0,
    [_DIPe],
    [() => DedicatedIpPool$]
];
exports.GetDedicatedIpPoolResponse$ = GetDedicatedIpPoolResponse$;
const GetDedicatedIpRequest$ = [3, n0, _GDIR,
    0,
    [_Ip],
    [[0, 1]], 1
];
exports.GetDedicatedIpRequest$ = GetDedicatedIpRequest$;
const GetDedicatedIpResponse$ = [3, n0, _GDIRe,
    0,
    [_DI],
    [() => DedicatedIp$]
];
exports.GetDedicatedIpResponse$ = GetDedicatedIpResponse$;
const GetDedicatedIpsRequest$ = [3, n0, _GDIRet,
    0,
    [_PN, _NT, _PS],
    [[0, { [_hQ]: _PN }], [0, { [_hQ]: _NT }], [1, { [_hQ]: _PS }]]
];
exports.GetDedicatedIpsRequest$ = GetDedicatedIpsRequest$;
const GetDedicatedIpsResponse$ = [3, n0, _GDIRete,
    0,
    [_DIe, _NT],
    [() => DedicatedIpList, 0]
];
exports.GetDedicatedIpsResponse$ = GetDedicatedIpsResponse$;
const GetDeliverabilityDashboardOptionsRequest$ = [3, n0, _GDDOR,
    0,
    [],
    []
];
exports.GetDeliverabilityDashboardOptionsRequest$ = GetDeliverabilityDashboardOptionsRequest$;
const GetDeliverabilityDashboardOptionsResponse$ = [3, n0, _GDDORe,
    0,
    [_DE, _SED, _AS, _ASD, _PESD],
    [2, 4, 0, () => DomainDeliverabilityTrackingOptions, () => DomainDeliverabilityTrackingOptions], 1
];
exports.GetDeliverabilityDashboardOptionsResponse$ = GetDeliverabilityDashboardOptionsResponse$;
const GetDeliverabilityTestReportRequest$ = [3, n0, _GDTRR,
    0,
    [_RI],
    [[0, 1]], 1
];
exports.GetDeliverabilityTestReportRequest$ = GetDeliverabilityTestReportRequest$;
const GetDeliverabilityTestReportResponse$ = [3, n0, _GDTRRe,
    0,
    [_DTRel, _OP, _IPs, _Me, _Ta],
    [() => DeliverabilityTestReport$, () => PlacementStatistics$, () => IspPlacements, 0, () => TagList], 3
];
exports.GetDeliverabilityTestReportResponse$ = GetDeliverabilityTestReportResponse$;
const GetDomainDeliverabilityCampaignRequest$ = [3, n0, _GDDCR,
    0,
    [_CIa],
    [[0, 1]], 1
];
exports.GetDomainDeliverabilityCampaignRequest$ = GetDomainDeliverabilityCampaignRequest$;
const GetDomainDeliverabilityCampaignResponse$ = [3, n0, _GDDCRe,
    0,
    [_DDC],
    [() => DomainDeliverabilityCampaign$], 1
];
exports.GetDomainDeliverabilityCampaignResponse$ = GetDomainDeliverabilityCampaignResponse$;
const GetDomainStatisticsReportRequest$ = [3, n0, _GDSRR,
    0,
    [_Do, _SD, _ED],
    [[0, 1], [4, { [_hQ]: _SD }], [4, { [_hQ]: _ED }]], 3
];
exports.GetDomainStatisticsReportRequest$ = GetDomainStatisticsReportRequest$;
const GetDomainStatisticsReportResponse$ = [3, n0, _GDSRRe,
    0,
    [_OV, _DVa],
    [() => OverallVolume$, () => DailyVolumes], 2
];
exports.GetDomainStatisticsReportResponse$ = GetDomainStatisticsReportResponse$;
const GetEmailAddressInsightsRequest$ = [3, n0, _GEAIR,
    0,
    [_EA],
    [0], 1
];
exports.GetEmailAddressInsightsRequest$ = GetEmailAddressInsightsRequest$;
const GetEmailAddressInsightsResponse$ = [3, n0, _GEAIRe,
    0,
    [_MV],
    [() => MailboxValidation$]
];
exports.GetEmailAddressInsightsResponse$ = GetEmailAddressInsightsResponse$;
const GetEmailIdentityPoliciesRequest$ = [3, n0, _GEIPR,
    0,
    [_EI],
    [[0, 1]], 1
];
exports.GetEmailIdentityPoliciesRequest$ = GetEmailIdentityPoliciesRequest$;
const GetEmailIdentityPoliciesResponse$ = [3, n0, _GEIPRe,
    0,
    [_Po],
    [128 | 0]
];
exports.GetEmailIdentityPoliciesResponse$ = GetEmailIdentityPoliciesResponse$;
const GetEmailIdentityRequest$ = [3, n0, _GEIR,
    0,
    [_EI],
    [[0, 1]], 1
];
exports.GetEmailIdentityRequest$ = GetEmailIdentityRequest$;
const GetEmailIdentityResponse$ = [3, n0, _GEIRe,
    0,
    [_IT, _FFS, _VFSS, _DA, _MFA, _Po, _Ta, _CSN, _VSe, _VI],
    [0, 2, 2, () => DkimAttributes$, () => MailFromAttributes$, 128 | 0, () => TagList, 0, 0, () => VerificationInfo$]
];
exports.GetEmailIdentityResponse$ = GetEmailIdentityResponse$;
const GetEmailTemplateRequest$ = [3, n0, _GETR,
    0,
    [_TN],
    [[0, 1]], 1
];
exports.GetEmailTemplateRequest$ = GetEmailTemplateRequest$;
const GetEmailTemplateResponse$ = [3, n0, _GETRe,
    0,
    [_TN, _TC, _Ta],
    [0, () => EmailTemplateContent$, () => TagList], 2
];
exports.GetEmailTemplateResponse$ = GetEmailTemplateResponse$;
const GetExportJobRequest$ = [3, n0, _GEJR,
    0,
    [_JI],
    [[0, 1]], 1
];
exports.GetExportJobRequest$ = GetExportJobRequest$;
const GetExportJobResponse$ = [3, n0, _GEJRe,
    0,
    [_JI, _EST, _JS, _EDx, _EDS, _CTr, _CTo, _FI, _St],
    [0, 0, 0, () => ExportDestination$, [() => ExportDataSource$, 0], 4, 4, () => FailureInfo$, () => ExportStatistics$]
];
exports.GetExportJobResponse$ = GetExportJobResponse$;
const GetImportJobRequest$ = [3, n0, _GIJR,
    0,
    [_JI],
    [[0, 1]], 1
];
exports.GetImportJobRequest$ = GetImportJobRequest$;
const GetImportJobResponse$ = [3, n0, _GIJRe,
    0,
    [_JI, _ID, _IDS, _FI, _JS, _CTr, _CTo, _PRC, _FRC],
    [0, () => ImportDestination$, () => ImportDataSource$, () => FailureInfo$, 0, 4, 4, 1, 1]
];
exports.GetImportJobResponse$ = GetImportJobResponse$;
const GetMessageInsightsRequest$ = [3, n0, _GMIR,
    0,
    [_MI],
    [[0, 1]], 1
];
exports.GetMessageInsightsRequest$ = GetMessageInsightsRequest$;
const GetMessageInsightsResponse$ = [3, n0, _GMIRe,
    0,
    [_MI, _FEA, _Su, _ET, _In],
    [0, [() => InsightsEmailAddress, 0], [() => EmailSubject, 0], () => MessageTagList, [() => EmailInsightsList, 0]]
];
exports.GetMessageInsightsResponse$ = GetMessageInsightsResponse$;
const GetMultiRegionEndpointRequest$ = [3, n0, _GMRER,
    0,
    [_EN],
    [[0, 1]], 1
];
exports.GetMultiRegionEndpointRequest$ = GetMultiRegionEndpointRequest$;
const GetMultiRegionEndpointResponse$ = [3, n0, _GMRERe,
    0,
    [_EN, _EIn, _Ro, _S, _CTr, _LUT],
    [0, 0, () => Routes, 0, 4, 4]
];
exports.GetMultiRegionEndpointResponse$ = GetMultiRegionEndpointResponse$;
const GetReputationEntityRequest$ = [3, n0, _GRER,
    0,
    [_RER, _RET],
    [[0, 1], [0, 1]], 2
];
exports.GetReputationEntityRequest$ = GetReputationEntityRequest$;
const GetReputationEntityResponse$ = [3, n0, _GRERe,
    0,
    [_RE],
    [() => ReputationEntity$]
];
exports.GetReputationEntityResponse$ = GetReputationEntityResponse$;
const GetSuppressedDestinationRequest$ = [3, n0, _GSDR,
    0,
    [_EA, _TNe],
    [[0, 1], [0, { [_hQ]: _TNe }]], 1
];
exports.GetSuppressedDestinationRequest$ = GetSuppressedDestinationRequest$;
const GetSuppressedDestinationResponse$ = [3, n0, _GSDRe,
    0,
    [_SDu],
    [() => SuppressedDestination$], 1
];
exports.GetSuppressedDestinationResponse$ = GetSuppressedDestinationResponse$;
const GetTenantRequest$ = [3, n0, _GTR,
    0,
    [_TNe],
    [0], 1
];
exports.GetTenantRequest$ = GetTenantRequest$;
const GetTenantResponse$ = [3, n0, _GTRe,
    0,
    [_Ten],
    [() => Tenant$]
];
exports.GetTenantResponse$ = GetTenantResponse$;
const GuardianAttributes$ = [3, n0, _GA,
    0,
    [_OSD],
    [0]
];
exports.GuardianAttributes$ = GuardianAttributes$;
const GuardianOptions$ = [3, n0, _GO,
    0,
    [_OSD],
    [0]
];
exports.GuardianOptions$ = GuardianOptions$;
const IdentityInfo$ = [3, n0, _II,
    0,
    [_IT, _INd, _SEe, _VSe],
    [0, 0, 2, 0]
];
exports.IdentityInfo$ = IdentityInfo$;
const ImportDataSource$ = [3, n0, _IDS,
    0,
    [_SU, _DF],
    [0, 0], 2
];
exports.ImportDataSource$ = ImportDataSource$;
const ImportDestination$ = [3, n0, _ID,
    0,
    [_SLD, _CLD],
    [() => SuppressionListDestination$, () => ContactListDestination$]
];
exports.ImportDestination$ = ImportDestination$;
const ImportJobSummary$ = [3, n0, _IJS,
    0,
    [_JI, _ID, _JS, _CTr, _PRC, _FRC],
    [0, () => ImportDestination$, 0, 4, 1, 1]
];
exports.ImportJobSummary$ = ImportJobSummary$;
const InboxPlacementTrackingOption$ = [3, n0, _IPTO,
    0,
    [_G, _TIr],
    [2, 64 | 0]
];
exports.InboxPlacementTrackingOption$ = InboxPlacementTrackingOption$;
const InsightsEvent$ = [3, n0, _IE,
    0,
    [_Ti, _Ty, _Det],
    [4, 0, () => EventDetails$]
];
exports.InsightsEvent$ = InsightsEvent$;
const IspPlacement$ = [3, n0, _IPsp,
    0,
    [_IN, _PSl],
    [0, () => PlacementStatistics$]
];
exports.IspPlacement$ = IspPlacement$;
const KinesisFirehoseDestination$ = [3, n0, _KFD,
    0,
    [_IRAa, _DSAe],
    [0, 0], 2
];
exports.KinesisFirehoseDestination$ = KinesisFirehoseDestination$;
const ListConfigurationSetsRequest$ = [3, n0, _LCSR,
    0,
    [_NT, _PS],
    [[0, { [_hQ]: _NT }], [1, { [_hQ]: _PS }]]
];
exports.ListConfigurationSetsRequest$ = ListConfigurationSetsRequest$;
const ListConfigurationSetsResponse$ = [3, n0, _LCSRi,
    0,
    [_CS, _NT],
    [64 | 0, 0]
];
exports.ListConfigurationSetsResponse$ = ListConfigurationSetsResponse$;
const ListContactListsRequest$ = [3, n0, _LCLR,
    0,
    [_PS, _NT],
    [[1, { [_hQ]: _PS }], [0, { [_hQ]: _NT }]]
];
exports.ListContactListsRequest$ = ListContactListsRequest$;
const ListContactListsResponse$ = [3, n0, _LCLRi,
    0,
    [_CLon, _NT],
    [() => ListOfContactLists, 0]
];
exports.ListContactListsResponse$ = ListContactListsResponse$;
const ListContactsFilter$ = [3, n0, _LCF,
    0,
    [_FS, _TF],
    [0, () => TopicFilter$]
];
exports.ListContactsFilter$ = ListContactsFilter$;
const ListContactsRequest$ = [3, n0, _LCR,
    0,
    [_CLN, _F, _PS, _NT],
    [[0, 1], () => ListContactsFilter$, 1, 0], 1
];
exports.ListContactsRequest$ = ListContactsRequest$;
const ListContactsResponse$ = [3, n0, _LCRi,
    0,
    [_Cont, _NT],
    [() => ListOfContacts, 0]
];
exports.ListContactsResponse$ = ListContactsResponse$;
const ListCustomVerificationEmailTemplatesRequest$ = [3, n0, _LCVETR,
    0,
    [_NT, _PS],
    [[0, { [_hQ]: _NT }], [1, { [_hQ]: _PS }]]
];
exports.ListCustomVerificationEmailTemplatesRequest$ = ListCustomVerificationEmailTemplatesRequest$;
const ListCustomVerificationEmailTemplatesResponse$ = [3, n0, _LCVETRi,
    0,
    [_CVET, _NT],
    [() => CustomVerificationEmailTemplatesList, 0]
];
exports.ListCustomVerificationEmailTemplatesResponse$ = ListCustomVerificationEmailTemplatesResponse$;
const ListDedicatedIpPoolsRequest$ = [3, n0, _LDIPR,
    0,
    [_NT, _PS],
    [[0, { [_hQ]: _NT }], [1, { [_hQ]: _PS }]]
];
exports.ListDedicatedIpPoolsRequest$ = ListDedicatedIpPoolsRequest$;
const ListDedicatedIpPoolsResponse$ = [3, n0, _LDIPRi,
    0,
    [_DIPed, _NT],
    [64 | 0, 0]
];
exports.ListDedicatedIpPoolsResponse$ = ListDedicatedIpPoolsResponse$;
const ListDeliverabilityTestReportsRequest$ = [3, n0, _LDTRR,
    0,
    [_NT, _PS],
    [[0, { [_hQ]: _NT }], [1, { [_hQ]: _PS }]]
];
exports.ListDeliverabilityTestReportsRequest$ = ListDeliverabilityTestReportsRequest$;
const ListDeliverabilityTestReportsResponse$ = [3, n0, _LDTRRi,
    0,
    [_DTReli, _NT],
    [() => DeliverabilityTestReports, 0], 1
];
exports.ListDeliverabilityTestReportsResponse$ = ListDeliverabilityTestReportsResponse$;
const ListDomainDeliverabilityCampaignsRequest$ = [3, n0, _LDDCR,
    0,
    [_SD, _ED, _SDub, _NT, _PS],
    [[4, { [_hQ]: _SD }], [4, { [_hQ]: _ED }], [0, 1], [0, { [_hQ]: _NT }], [1, { [_hQ]: _PS }]], 3
];
exports.ListDomainDeliverabilityCampaignsRequest$ = ListDomainDeliverabilityCampaignsRequest$;
const ListDomainDeliverabilityCampaignsResponse$ = [3, n0, _LDDCRi,
    0,
    [_DDCo, _NT],
    [() => DomainDeliverabilityCampaignList, 0], 1
];
exports.ListDomainDeliverabilityCampaignsResponse$ = ListDomainDeliverabilityCampaignsResponse$;
const ListEmailIdentitiesRequest$ = [3, n0, _LEIR,
    0,
    [_NT, _PS],
    [[0, { [_hQ]: _NT }], [1, { [_hQ]: _PS }]]
];
exports.ListEmailIdentitiesRequest$ = ListEmailIdentitiesRequest$;
const ListEmailIdentitiesResponse$ = [3, n0, _LEIRi,
    0,
    [_EIma, _NT],
    [() => IdentityInfoList, 0]
];
exports.ListEmailIdentitiesResponse$ = ListEmailIdentitiesResponse$;
const ListEmailTemplatesRequest$ = [3, n0, _LETR,
    0,
    [_NT, _PS],
    [[0, { [_hQ]: _NT }], [1, { [_hQ]: _PS }]]
];
exports.ListEmailTemplatesRequest$ = ListEmailTemplatesRequest$;
const ListEmailTemplatesResponse$ = [3, n0, _LETRi,
    0,
    [_TM, _NT],
    [() => EmailTemplateMetadataList, 0]
];
exports.ListEmailTemplatesResponse$ = ListEmailTemplatesResponse$;
const ListExportJobsRequest$ = [3, n0, _LEJR,
    0,
    [_NT, _PS, _EST, _JS],
    [0, 1, 0, 0]
];
exports.ListExportJobsRequest$ = ListExportJobsRequest$;
const ListExportJobsResponse$ = [3, n0, _LEJRi,
    0,
    [_EJ, _NT],
    [() => ExportJobSummaryList, 0]
];
exports.ListExportJobsResponse$ = ListExportJobsResponse$;
const ListImportJobsRequest$ = [3, n0, _LIJR,
    0,
    [_IDT, _NT, _PS],
    [0, 0, 1]
];
exports.ListImportJobsRequest$ = ListImportJobsRequest$;
const ListImportJobsResponse$ = [3, n0, _LIJRi,
    0,
    [_IJ, _NT],
    [() => ImportJobSummaryList, 0]
];
exports.ListImportJobsResponse$ = ListImportJobsResponse$;
const ListManagementOptions$ = [3, n0, _LMO,
    0,
    [_CLN, _TNo],
    [0, 0], 1
];
exports.ListManagementOptions$ = ListManagementOptions$;
const ListMultiRegionEndpointsRequest$ = [3, n0, _LMRER,
    0,
    [_NT, _PS],
    [[0, { [_hQ]: _NT }], [1, { [_hQ]: _PS }]]
];
exports.ListMultiRegionEndpointsRequest$ = ListMultiRegionEndpointsRequest$;
const ListMultiRegionEndpointsResponse$ = [3, n0, _LMRERi,
    0,
    [_MRE, _NT],
    [() => MultiRegionEndpoints, 0]
];
exports.ListMultiRegionEndpointsResponse$ = ListMultiRegionEndpointsResponse$;
const ListRecommendationsRequest$ = [3, n0, _LRR,
    0,
    [_F, _NT, _PS],
    [128 | 0, 0, 1]
];
exports.ListRecommendationsRequest$ = ListRecommendationsRequest$;
const ListRecommendationsResponse$ = [3, n0, _LRRi,
    0,
    [_Re, _NT],
    [() => RecommendationsList, 0]
];
exports.ListRecommendationsResponse$ = ListRecommendationsResponse$;
const ListReputationEntitiesRequest$ = [3, n0, _LRER,
    0,
    [_F, _NT, _PS],
    [128 | 0, 0, 1]
];
exports.ListReputationEntitiesRequest$ = ListReputationEntitiesRequest$;
const ListReputationEntitiesResponse$ = [3, n0, _LRERi,
    0,
    [_REe, _NT],
    [() => ReputationEntitiesList, 0]
];
exports.ListReputationEntitiesResponse$ = ListReputationEntitiesResponse$;
const ListResourceTenantsRequest$ = [3, n0, _LRTR,
    0,
    [_RA, _PS, _NT],
    [0, 1, 0], 1
];
exports.ListResourceTenantsRequest$ = ListResourceTenantsRequest$;
const ListResourceTenantsResponse$ = [3, n0, _LRTRi,
    0,
    [_RTe, _NT],
    [() => ResourceTenantMetadataList, 0]
];
exports.ListResourceTenantsResponse$ = ListResourceTenantsResponse$;
const ListSuppressedDestinationsRequest$ = [3, n0, _LSDR,
    0,
    [_TNe, _Rea, _SD, _ED, _NT, _PS],
    [[0, { [_hQ]: _TNe }], [64 | 0, { [_hQ]: _Reas }], [4, { [_hQ]: _SD }], [4, { [_hQ]: _ED }], [0, { [_hQ]: _NT }], [1, { [_hQ]: _PS }]]
];
exports.ListSuppressedDestinationsRequest$ = ListSuppressedDestinationsRequest$;
const ListSuppressedDestinationsResponse$ = [3, n0, _LSDRi,
    0,
    [_SDS, _NT],
    [() => SuppressedDestinationSummaries, 0]
];
exports.ListSuppressedDestinationsResponse$ = ListSuppressedDestinationsResponse$;
const ListTagsForResourceRequest$ = [3, n0, _LTFRR,
    0,
    [_RA],
    [[0, { [_hQ]: _RA }]], 1
];
exports.ListTagsForResourceRequest$ = ListTagsForResourceRequest$;
const ListTagsForResourceResponse$ = [3, n0, _LTFRRi,
    0,
    [_Ta],
    [() => TagList], 1
];
exports.ListTagsForResourceResponse$ = ListTagsForResourceResponse$;
const ListTenantResourcesRequest$ = [3, n0, _LTRR,
    0,
    [_TNe, _F, _PS, _NT],
    [0, 128 | 0, 1, 0], 1
];
exports.ListTenantResourcesRequest$ = ListTenantResourcesRequest$;
const ListTenantResourcesResponse$ = [3, n0, _LTRRi,
    0,
    [_TR, _NT],
    [() => TenantResourceList, 0]
];
exports.ListTenantResourcesResponse$ = ListTenantResourcesResponse$;
const ListTenantsRequest$ = [3, n0, _LTR,
    0,
    [_NT, _PS],
    [0, 1]
];
exports.ListTenantsRequest$ = ListTenantsRequest$;
const ListTenantsResponse$ = [3, n0, _LTRi,
    0,
    [_Tena, _NT],
    [() => TenantInfoList, 0]
];
exports.ListTenantsResponse$ = ListTenantsResponse$;
const MailboxValidation$ = [3, n0, _MV,
    0,
    [_IV, _Eva],
    [() => EmailAddressInsightsVerdict$, () => EmailAddressInsightsMailboxEvaluations$]
];
exports.MailboxValidation$ = MailboxValidation$;
const MailFromAttributes$ = [3, n0, _MFA,
    0,
    [_MFD, _MFDS, _BOMF],
    [0, 0, 0], 3
];
exports.MailFromAttributes$ = MailFromAttributes$;
const Message$ = [3, n0, _Me,
    0,
    [_Su, _B, _He, _At],
    [() => Content$, () => Body$, () => MessageHeaderList, () => AttachmentList], 2
];
exports.Message$ = Message$;
const MessageHeader$ = [3, n0, _MH,
    0,
    [_Na, _V],
    [0, 0], 2
];
exports.MessageHeader$ = MessageHeader$;
const MessageInsightsDataSource$ = [3, n0, _MIDS,
    0,
    [_SD, _ED, _Inc, _Ex, _MRa],
    [4, 4, [() => MessageInsightsFilters$, 0], [() => MessageInsightsFilters$, 0], 1], 2
];
exports.MessageInsightsDataSource$ = MessageInsightsDataSource$;
const MessageInsightsFilters$ = [3, n0, _MIF,
    0,
    [_FEA, _Des, _Su, _Is, _LDE, _LEEa],
    [[() => EmailAddressFilterList, 0], [() => EmailAddressFilterList, 0], [() => EmailSubjectFilterList, 0], 64 | 0, 64 | 0, 64 | 0]
];
exports.MessageInsightsFilters$ = MessageInsightsFilters$;
const MessageTag$ = [3, n0, _MTe,
    0,
    [_Na, _V],
    [0, 0], 2
];
exports.MessageTag$ = MessageTag$;
const MetricDataError$ = [3, n0, _MDE,
    0,
    [_I, _Cod, _Me],
    [0, 0, 0]
];
exports.MetricDataError$ = MetricDataError$;
const MetricDataResult$ = [3, n0, _MDR,
    0,
    [_I, _Tim, _Va],
    [0, 64 | 4, 64 | 1]
];
exports.MetricDataResult$ = MetricDataResult$;
const MetricsDataSource$ = [3, n0, _MDSe,
    0,
    [_D, _N, _Met, _SD, _ED],
    [[2, n0, _EDxp, 0, 0, 64 | 0], 0, () => ExportMetrics, 4, 4], 5
];
exports.MetricsDataSource$ = MetricsDataSource$;
const MultiRegionEndpoint$ = [3, n0, _MREu,
    0,
    [_EN, _S, _EIn, _Reg, _CTr, _LUT],
    [0, 0, 0, 64 | 0, 4, 4]
];
exports.MultiRegionEndpoint$ = MultiRegionEndpoint$;
const OverallVolume$ = [3, n0, _OV,
    0,
    [_VS, _RRP, _DIP],
    [() => VolumeStatistics$, 1, () => DomainIspPlacements]
];
exports.OverallVolume$ = OverallVolume$;
const PinpointDestination$ = [3, n0, _PD,
    0,
    [_AAp],
    [0]
];
exports.PinpointDestination$ = PinpointDestination$;
const PlacementStatistics$ = [3, n0, _PSl,
    0,
    [_IP, _SP, _MP, _SPp, _DP],
    [1, 1, 1, 1, 1]
];
exports.PlacementStatistics$ = PlacementStatistics$;
const PutAccountDedicatedIpWarmupAttributesRequest$ = [3, n0, _PADIWAR,
    0,
    [_AWE],
    [2]
];
exports.PutAccountDedicatedIpWarmupAttributesRequest$ = PutAccountDedicatedIpWarmupAttributesRequest$;
const PutAccountDedicatedIpWarmupAttributesResponse$ = [3, n0, _PADIWARu,
    0,
    [],
    []
];
exports.PutAccountDedicatedIpWarmupAttributesResponse$ = PutAccountDedicatedIpWarmupAttributesResponse$;
const PutAccountDetailsRequest$ = [3, n0, _PADR,
    0,
    [_MT, _WURL, _CL, _UCD, _ACEAd, _PAE],
    [0, [() => WebsiteURL, 0], 0, [() => UseCaseDescription, 0], [() => AdditionalContactEmailAddresses, 0], 2], 2
];
exports.PutAccountDetailsRequest$ = PutAccountDetailsRequest$;
const PutAccountDetailsResponse$ = [3, n0, _PADRu,
    0,
    [],
    []
];
exports.PutAccountDetailsResponse$ = PutAccountDetailsResponse$;
const PutAccountSendingAttributesRequest$ = [3, n0, _PASAR,
    0,
    [_SEe],
    [2]
];
exports.PutAccountSendingAttributesRequest$ = PutAccountSendingAttributesRequest$;
const PutAccountSendingAttributesResponse$ = [3, n0, _PASARu,
    0,
    [],
    []
];
exports.PutAccountSendingAttributesResponse$ = PutAccountSendingAttributesResponse$;
const PutAccountSuppressionAttributesRequest$ = [3, n0, _PASARut,
    0,
    [_SR, _VAa],
    [64 | 0, () => SuppressionValidationAttributes$]
];
exports.PutAccountSuppressionAttributesRequest$ = PutAccountSuppressionAttributesRequest$;
const PutAccountSuppressionAttributesResponse$ = [3, n0, _PASARutc,
    0,
    [],
    []
];
exports.PutAccountSuppressionAttributesResponse$ = PutAccountSuppressionAttributesResponse$;
const PutAccountVdmAttributesRequest$ = [3, n0, _PAVAR,
    0,
    [_VA],
    [() => VdmAttributes$], 1
];
exports.PutAccountVdmAttributesRequest$ = PutAccountVdmAttributesRequest$;
const PutAccountVdmAttributesResponse$ = [3, n0, _PAVARu,
    0,
    [],
    []
];
exports.PutAccountVdmAttributesResponse$ = PutAccountVdmAttributesResponse$;
const PutConfigurationSetArchivingOptionsRequest$ = [3, n0, _PCSAOR,
    0,
    [_CSN, _AA],
    [[0, 1], 0], 1
];
exports.PutConfigurationSetArchivingOptionsRequest$ = PutConfigurationSetArchivingOptionsRequest$;
const PutConfigurationSetArchivingOptionsResponse$ = [3, n0, _PCSAORu,
    0,
    [],
    []
];
exports.PutConfigurationSetArchivingOptionsResponse$ = PutConfigurationSetArchivingOptionsResponse$;
const PutConfigurationSetDeliveryOptionsRequest$ = [3, n0, _PCSDOR,
    0,
    [_CSN, _TPl, _SPN, _MDS],
    [[0, 1], 0, 0, 1], 1
];
exports.PutConfigurationSetDeliveryOptionsRequest$ = PutConfigurationSetDeliveryOptionsRequest$;
const PutConfigurationSetDeliveryOptionsResponse$ = [3, n0, _PCSDORu,
    0,
    [],
    []
];
exports.PutConfigurationSetDeliveryOptionsResponse$ = PutConfigurationSetDeliveryOptionsResponse$;
const PutConfigurationSetReputationOptionsRequest$ = [3, n0, _PCSROR,
    0,
    [_CSN, _RME],
    [[0, 1], 2], 1
];
exports.PutConfigurationSetReputationOptionsRequest$ = PutConfigurationSetReputationOptionsRequest$;
const PutConfigurationSetReputationOptionsResponse$ = [3, n0, _PCSRORu,
    0,
    [],
    []
];
exports.PutConfigurationSetReputationOptionsResponse$ = PutConfigurationSetReputationOptionsResponse$;
const PutConfigurationSetSendingOptionsRequest$ = [3, n0, _PCSSOR,
    0,
    [_CSN, _SEe],
    [[0, 1], 2], 1
];
exports.PutConfigurationSetSendingOptionsRequest$ = PutConfigurationSetSendingOptionsRequest$;
const PutConfigurationSetSendingOptionsResponse$ = [3, n0, _PCSSORu,
    0,
    [],
    []
];
exports.PutConfigurationSetSendingOptionsResponse$ = PutConfigurationSetSendingOptionsResponse$;
const PutConfigurationSetSuppressionOptionsRequest$ = [3, n0, _PCSSORut,
    0,
    [_CSN, _SSu, _SR, _VOa],
    [[0, 1], 0, 64 | 0, () => SuppressionValidationOptions$], 1
];
exports.PutConfigurationSetSuppressionOptionsRequest$ = PutConfigurationSetSuppressionOptionsRequest$;
const PutConfigurationSetSuppressionOptionsResponse$ = [3, n0, _PCSSORuto,
    0,
    [],
    []
];
exports.PutConfigurationSetSuppressionOptionsResponse$ = PutConfigurationSetSuppressionOptionsResponse$;
const PutConfigurationSetTrackingOptionsRequest$ = [3, n0, _PCSTOR,
    0,
    [_CSN, _CRD, _HP],
    [[0, 1], 0, 0], 1
];
exports.PutConfigurationSetTrackingOptionsRequest$ = PutConfigurationSetTrackingOptionsRequest$;
const PutConfigurationSetTrackingOptionsResponse$ = [3, n0, _PCSTORu,
    0,
    [],
    []
];
exports.PutConfigurationSetTrackingOptionsResponse$ = PutConfigurationSetTrackingOptionsResponse$;
const PutConfigurationSetVdmOptionsRequest$ = [3, n0, _PCSVOR,
    0,
    [_CSN, _VO],
    [[0, 1], () => VdmOptions$], 1
];
exports.PutConfigurationSetVdmOptionsRequest$ = PutConfigurationSetVdmOptionsRequest$;
const PutConfigurationSetVdmOptionsResponse$ = [3, n0, _PCSVORu,
    0,
    [],
    []
];
exports.PutConfigurationSetVdmOptionsResponse$ = PutConfigurationSetVdmOptionsResponse$;
const PutDedicatedIpInPoolRequest$ = [3, n0, _PDIIPR,
    0,
    [_Ip, _DPN],
    [[0, 1], 0], 2
];
exports.PutDedicatedIpInPoolRequest$ = PutDedicatedIpInPoolRequest$;
const PutDedicatedIpInPoolResponse$ = [3, n0, _PDIIPRu,
    0,
    [],
    []
];
exports.PutDedicatedIpInPoolResponse$ = PutDedicatedIpInPoolResponse$;
const PutDedicatedIpPoolScalingAttributesRequest$ = [3, n0, _PDIPSAR,
    0,
    [_PN, _SM],
    [[0, 1], 0], 2
];
exports.PutDedicatedIpPoolScalingAttributesRequest$ = PutDedicatedIpPoolScalingAttributesRequest$;
const PutDedicatedIpPoolScalingAttributesResponse$ = [3, n0, _PDIPSARu,
    0,
    [],
    []
];
exports.PutDedicatedIpPoolScalingAttributesResponse$ = PutDedicatedIpPoolScalingAttributesResponse$;
const PutDedicatedIpWarmupAttributesRequest$ = [3, n0, _PDIWAR,
    0,
    [_Ip, _WP],
    [[0, 1], 1], 2
];
exports.PutDedicatedIpWarmupAttributesRequest$ = PutDedicatedIpWarmupAttributesRequest$;
const PutDedicatedIpWarmupAttributesResponse$ = [3, n0, _PDIWARu,
    0,
    [],
    []
];
exports.PutDedicatedIpWarmupAttributesResponse$ = PutDedicatedIpWarmupAttributesResponse$;
const PutDeliverabilityDashboardOptionRequest$ = [3, n0, _PDDOR,
    0,
    [_DE, _SDubs],
    [2, () => DomainDeliverabilityTrackingOptions], 1
];
exports.PutDeliverabilityDashboardOptionRequest$ = PutDeliverabilityDashboardOptionRequest$;
const PutDeliverabilityDashboardOptionResponse$ = [3, n0, _PDDORu,
    0,
    [],
    []
];
exports.PutDeliverabilityDashboardOptionResponse$ = PutDeliverabilityDashboardOptionResponse$;
const PutEmailIdentityConfigurationSetAttributesRequest$ = [3, n0, _PEICSAR,
    0,
    [_EI, _CSN],
    [[0, 1], 0], 1
];
exports.PutEmailIdentityConfigurationSetAttributesRequest$ = PutEmailIdentityConfigurationSetAttributesRequest$;
const PutEmailIdentityConfigurationSetAttributesResponse$ = [3, n0, _PEICSARu,
    0,
    [],
    []
];
exports.PutEmailIdentityConfigurationSetAttributesResponse$ = PutEmailIdentityConfigurationSetAttributesResponse$;
const PutEmailIdentityDkimAttributesRequest$ = [3, n0, _PEIDAR,
    0,
    [_EI, _SE],
    [[0, 1], 2], 1
];
exports.PutEmailIdentityDkimAttributesRequest$ = PutEmailIdentityDkimAttributesRequest$;
const PutEmailIdentityDkimAttributesResponse$ = [3, n0, _PEIDARu,
    0,
    [],
    []
];
exports.PutEmailIdentityDkimAttributesResponse$ = PutEmailIdentityDkimAttributesResponse$;
const PutEmailIdentityDkimSigningAttributesRequest$ = [3, n0, _PEIDSAR,
    0,
    [_EI, _SAO, _SAi],
    [[0, 1], 0, [() => DkimSigningAttributes$, 0]], 2
];
exports.PutEmailIdentityDkimSigningAttributesRequest$ = PutEmailIdentityDkimSigningAttributesRequest$;
const PutEmailIdentityDkimSigningAttributesResponse$ = [3, n0, _PEIDSARu,
    0,
    [_DS, _DT, _SHZ],
    [0, 64 | 0, 0]
];
exports.PutEmailIdentityDkimSigningAttributesResponse$ = PutEmailIdentityDkimSigningAttributesResponse$;
const PutEmailIdentityFeedbackAttributesRequest$ = [3, n0, _PEIFAR,
    0,
    [_EI, _EFE],
    [[0, 1], 2], 1
];
exports.PutEmailIdentityFeedbackAttributesRequest$ = PutEmailIdentityFeedbackAttributesRequest$;
const PutEmailIdentityFeedbackAttributesResponse$ = [3, n0, _PEIFARu,
    0,
    [],
    []
];
exports.PutEmailIdentityFeedbackAttributesResponse$ = PutEmailIdentityFeedbackAttributesResponse$;
const PutEmailIdentityMailFromAttributesRequest$ = [3, n0, _PEIMFAR,
    0,
    [_EI, _MFD, _BOMF],
    [[0, 1], 0, 0], 1
];
exports.PutEmailIdentityMailFromAttributesRequest$ = PutEmailIdentityMailFromAttributesRequest$;
const PutEmailIdentityMailFromAttributesResponse$ = [3, n0, _PEIMFARu,
    0,
    [],
    []
];
exports.PutEmailIdentityMailFromAttributesResponse$ = PutEmailIdentityMailFromAttributesResponse$;
const PutSuppressedDestinationRequest$ = [3, n0, _PSDR,
    0,
    [_EA, _Reas, _TNe],
    [0, 0, 0], 2
];
exports.PutSuppressedDestinationRequest$ = PutSuppressedDestinationRequest$;
const PutSuppressedDestinationResponse$ = [3, n0, _PSDRu,
    0,
    [],
    []
];
exports.PutSuppressedDestinationResponse$ = PutSuppressedDestinationResponse$;
const PutTenantSuppressionAttributesRequest$ = [3, n0, _PTSAR,
    0,
    [_TNe, _SR, _SSu],
    [0, 64 | 0, 0], 1
];
exports.PutTenantSuppressionAttributesRequest$ = PutTenantSuppressionAttributesRequest$;
const PutTenantSuppressionAttributesResponse$ = [3, n0, _PTSARu,
    0,
    [],
    []
];
exports.PutTenantSuppressionAttributesResponse$ = PutTenantSuppressionAttributesResponse$;
const RawMessage$ = [3, n0, _RM,
    0,
    [_Da],
    [21], 1
];
exports.RawMessage$ = RawMessage$;
const Recommendation$ = [3, n0, _Rec,
    0,
    [_RA, _Ty, _De, _S, _CTr, _LUT, _Im],
    [0, 0, 0, 0, 4, 4, 0]
];
exports.Recommendation$ = Recommendation$;
const ReplacementEmailContent$ = [3, n0, _REC,
    0,
    [_RTep],
    [() => ReplacementTemplate$]
];
exports.ReplacementEmailContent$ = ReplacementEmailContent$;
const ReplacementTemplate$ = [3, n0, _RTep,
    0,
    [_RTD],
    [0]
];
exports.ReplacementTemplate$ = ReplacementTemplate$;
const ReputationEntity$ = [3, n0, _RE,
    0,
    [_RER, _RET, _RMP, _CMS, _ASMS, _SSA, _RIe],
    [0, 0, 0, () => StatusRecord$, () => StatusRecord$, 0, 0]
];
exports.ReputationEntity$ = ReputationEntity$;
const ReputationOptions$ = [3, n0, _RO,
    0,
    [_RME, _LFS],
    [2, 4]
];
exports.ReputationOptions$ = ReputationOptions$;
const ResourceTenantMetadata$ = [3, n0, _RTM,
    0,
    [_TNe, _TI, _RA, _AT],
    [0, 0, 0, 4]
];
exports.ResourceTenantMetadata$ = ResourceTenantMetadata$;
const ReviewDetails$ = [3, n0, _RD,
    0,
    [_S, _CIas],
    [0, 0]
];
exports.ReviewDetails$ = ReviewDetails$;
const Route$ = [3, n0, _Rou,
    0,
    [_Regi],
    [0], 1
];
exports.Route$ = Route$;
const RouteDetails$ = [3, n0, _RDou,
    0,
    [_Regi],
    [0], 1
];
exports.RouteDetails$ = RouteDetails$;
const SendBulkEmailRequest$ = [3, n0, _SBER,
    0,
    [_DCe, _BEEu, _FEA, _FEAIA, _RTA, _FFEA, _FFEAIA, _DET, _CSN, _EIn, _TNe],
    [() => BulkEmailContent$, () => BulkEmailEntryList, 0, 0, 64 | 0, 0, 0, () => MessageTagList, 0, 0, 0], 2
];
exports.SendBulkEmailRequest$ = SendBulkEmailRequest$;
const SendBulkEmailResponse$ = [3, n0, _SBERe,
    0,
    [_BEERu],
    [() => BulkEmailEntryResultList], 1
];
exports.SendBulkEmailResponse$ = SendBulkEmailResponse$;
const SendCustomVerificationEmailRequest$ = [3, n0, _SCVER,
    0,
    [_EA, _TN, _CSN],
    [0, 0, 0], 2
];
exports.SendCustomVerificationEmailRequest$ = SendCustomVerificationEmailRequest$;
const SendCustomVerificationEmailResponse$ = [3, n0, _SCVERe,
    0,
    [_MI],
    [0]
];
exports.SendCustomVerificationEmailResponse$ = SendCustomVerificationEmailResponse$;
const SendEmailRequest$ = [3, n0, _SER,
    0,
    [_Con, _FEA, _FEAIA, _Des, _RTA, _FFEA, _FFEAIA, _ET, _CSN, _EIn, _TNe, _LMO],
    [() => EmailContent$, 0, 0, () => Destination$, 64 | 0, 0, 0, () => MessageTagList, 0, 0, 0, () => ListManagementOptions$], 1
];
exports.SendEmailRequest$ = SendEmailRequest$;
const SendEmailResponse$ = [3, n0, _SERe,
    0,
    [_MI],
    [0]
];
exports.SendEmailResponse$ = SendEmailResponse$;
const SendingOptions$ = [3, n0, _SO,
    0,
    [_SEe],
    [2]
];
exports.SendingOptions$ = SendingOptions$;
const SendQuota$ = [3, n0, _SQ,
    0,
    [_MHS, _MSR, _SLH],
    [1, 1, 1]
];
exports.SendQuota$ = SendQuota$;
const SnsDestination$ = [3, n0, _SDn,
    0,
    [_TAop],
    [0], 1
];
exports.SnsDestination$ = SnsDestination$;
const SOARecord$ = [3, n0, _SOAR,
    0,
    [_PNS, _AE, _SN],
    [0, 0, 1]
];
exports.SOARecord$ = SOARecord$;
const StatusRecord$ = [3, n0, _SRt,
    0,
    [_S, _Ca, _LUT],
    [0, 0, 4]
];
exports.StatusRecord$ = StatusRecord$;
const SuppressedDestination$ = [3, n0, _SDu,
    0,
    [_EA, _Reas, _LUTa, _Att, _TNe],
    [0, 0, 4, () => SuppressedDestinationAttributes$, 0], 3
];
exports.SuppressedDestination$ = SuppressedDestination$;
const SuppressedDestinationAttributes$ = [3, n0, _SDA,
    0,
    [_MI, _FIe],
    [0, 0]
];
exports.SuppressedDestinationAttributes$ = SuppressedDestinationAttributes$;
const SuppressedDestinationSummary$ = [3, n0, _SDSu,
    0,
    [_EA, _Reas, _LUTa],
    [0, 0, 4], 3
];
exports.SuppressedDestinationSummary$ = SuppressedDestinationSummary$;
const SuppressionAttributes$ = [3, n0, _SA,
    0,
    [_SR, _VAa],
    [64 | 0, () => SuppressionValidationAttributes$]
];
exports.SuppressionAttributes$ = SuppressionAttributes$;
const SuppressionConditionThreshold$ = [3, n0, _SCT,
    0,
    [_CTEo, _OCT],
    [0, () => SuppressionConfidenceThreshold$], 1
];
exports.SuppressionConditionThreshold$ = SuppressionConditionThreshold$;
const SuppressionConfidenceThreshold$ = [3, n0, _SCTu,
    0,
    [_CVT],
    [0], 1
];
exports.SuppressionConfidenceThreshold$ = SuppressionConfidenceThreshold$;
const SuppressionListDestination$ = [3, n0, _SLD,
    0,
    [_SLIA],
    [0], 1
];
exports.SuppressionListDestination$ = SuppressionListDestination$;
const SuppressionOptions$ = [3, n0, _SOu,
    0,
    [_SR, _SSu, _VOa],
    [64 | 0, 0, () => SuppressionValidationOptions$]
];
exports.SuppressionOptions$ = SuppressionOptions$;
const SuppressionValidationAttributes$ = [3, n0, _SVA,
    0,
    [_CTon],
    [() => SuppressionConditionThreshold$], 1
];
exports.SuppressionValidationAttributes$ = SuppressionValidationAttributes$;
const SuppressionValidationOptions$ = [3, n0, _SVO,
    0,
    [_CTon],
    [() => SuppressionConditionThreshold$], 1
];
exports.SuppressionValidationOptions$ = SuppressionValidationOptions$;
const Tag$ = [3, n0, _Tag,
    0,
    [_K, _V],
    [0, 0], 2
];
exports.Tag$ = Tag$;
const TagResourceRequest$ = [3, n0, _TRR,
    0,
    [_RA, _Ta],
    [0, () => TagList], 2
];
exports.TagResourceRequest$ = TagResourceRequest$;
const TagResourceResponse$ = [3, n0, _TRRa,
    0,
    [],
    []
];
exports.TagResourceResponse$ = TagResourceResponse$;
const Template$ = [3, n0, _Te,
    0,
    [_TN, _TAe, _TC, _TD, _He, _At],
    [0, 0, () => EmailTemplateContent$, 0, () => MessageHeaderList, () => AttachmentList]
];
exports.Template$ = Template$;
const Tenant$ = [3, n0, _Ten,
    0,
    [_TNe, _TI, _TA, _CTr, _Ta, _SS, _SA],
    [0, 0, 0, 4, () => TagList, 0, () => TenantSuppressionAttributes$]
];
exports.Tenant$ = Tenant$;
const TenantInfo$ = [3, n0, _TIe,
    0,
    [_TNe, _TI, _TA, _CTr],
    [0, 0, 0, 4]
];
exports.TenantInfo$ = TenantInfo$;
const TenantResource$ = [3, n0, _TRe,
    0,
    [_RTes, _RA],
    [0, 0]
];
exports.TenantResource$ = TenantResource$;
const TenantSuppressionAttributes$ = [3, n0, _TSA,
    0,
    [_SR, _SSu],
    [64 | 0, 0]
];
exports.TenantSuppressionAttributes$ = TenantSuppressionAttributes$;
const TestRenderEmailTemplateRequest$ = [3, n0, _TRETR,
    0,
    [_TN, _TD],
    [[0, 1], 0], 2
];
exports.TestRenderEmailTemplateRequest$ = TestRenderEmailTemplateRequest$;
const TestRenderEmailTemplateResponse$ = [3, n0, _TRETRe,
    0,
    [_RTen],
    [0], 1
];
exports.TestRenderEmailTemplateResponse$ = TestRenderEmailTemplateResponse$;
const Topic$ = [3, n0, _Top,
    0,
    [_TNo, _DNi, _DSSe, _De],
    [0, 0, 0, 0], 3
];
exports.Topic$ = Topic$;
const TopicFilter$ = [3, n0, _TF,
    0,
    [_TNo, _UDIPU],
    [0, 2]
];
exports.TopicFilter$ = TopicFilter$;
const TopicPreference$ = [3, n0, _TPo,
    0,
    [_TNo, _SSub],
    [0, 0], 2
];
exports.TopicPreference$ = TopicPreference$;
const TrackingOptions$ = [3, n0, _TO,
    0,
    [_CRD, _HP],
    [0, 0], 1
];
exports.TrackingOptions$ = TrackingOptions$;
const UntagResourceRequest$ = [3, n0, _URR,
    0,
    [_RA, _TK],
    [[0, { [_hQ]: _RA }], [64 | 0, { [_hQ]: _TK }]], 2
];
exports.UntagResourceRequest$ = UntagResourceRequest$;
const UntagResourceResponse$ = [3, n0, _URRn,
    0,
    [],
    []
];
exports.UntagResourceResponse$ = UntagResourceResponse$;
const UpdateConfigurationSetEventDestinationRequest$ = [3, n0, _UCSEDR,
    0,
    [_CSN, _EDN, _EDv],
    [[0, 1], [0, 1], () => EventDestinationDefinition$], 3
];
exports.UpdateConfigurationSetEventDestinationRequest$ = UpdateConfigurationSetEventDestinationRequest$;
const UpdateConfigurationSetEventDestinationResponse$ = [3, n0, _UCSEDRp,
    0,
    [],
    []
];
exports.UpdateConfigurationSetEventDestinationResponse$ = UpdateConfigurationSetEventDestinationResponse$;
const UpdateContactListRequest$ = [3, n0, _UCLR,
    0,
    [_CLN, _To, _De],
    [[0, 1], () => Topics, 0], 1
];
exports.UpdateContactListRequest$ = UpdateContactListRequest$;
const UpdateContactListResponse$ = [3, n0, _UCLRp,
    0,
    [],
    []
];
exports.UpdateContactListResponse$ = UpdateContactListResponse$;
const UpdateContactRequest$ = [3, n0, _UCR,
    0,
    [_CLN, _EA, _TP, _UA, _ADt],
    [[0, 1], [0, 1], () => TopicPreferenceList, 2, 0], 2
];
exports.UpdateContactRequest$ = UpdateContactRequest$;
const UpdateContactResponse$ = [3, n0, _UCRp,
    0,
    [],
    []
];
exports.UpdateContactResponse$ = UpdateContactResponse$;
const UpdateCustomVerificationEmailTemplateRequest$ = [3, n0, _UCVETR,
    0,
    [_TN, _FEA, _TS, _TC, _SRURL, _FRURL],
    [[0, 1], 0, 0, 0, 0, 0], 6
];
exports.UpdateCustomVerificationEmailTemplateRequest$ = UpdateCustomVerificationEmailTemplateRequest$;
const UpdateCustomVerificationEmailTemplateResponse$ = [3, n0, _UCVETRp,
    0,
    [],
    []
];
exports.UpdateCustomVerificationEmailTemplateResponse$ = UpdateCustomVerificationEmailTemplateResponse$;
const UpdateEmailIdentityPolicyRequest$ = [3, n0, _UEIPR,
    0,
    [_EI, _PNo, _P],
    [[0, 1], [0, 1], 0], 3
];
exports.UpdateEmailIdentityPolicyRequest$ = UpdateEmailIdentityPolicyRequest$;
const UpdateEmailIdentityPolicyResponse$ = [3, n0, _UEIPRp,
    0,
    [],
    []
];
exports.UpdateEmailIdentityPolicyResponse$ = UpdateEmailIdentityPolicyResponse$;
const UpdateEmailTemplateRequest$ = [3, n0, _UETR,
    0,
    [_TN, _TC],
    [[0, 1], () => EmailTemplateContent$], 2
];
exports.UpdateEmailTemplateRequest$ = UpdateEmailTemplateRequest$;
const UpdateEmailTemplateResponse$ = [3, n0, _UETRp,
    0,
    [],
    []
];
exports.UpdateEmailTemplateResponse$ = UpdateEmailTemplateResponse$;
const UpdateReputationEntityCustomerManagedStatusRequest$ = [3, n0, _URECMSR,
    0,
    [_RET, _RER, _SS],
    [[0, 1], [0, 1], 0], 3
];
exports.UpdateReputationEntityCustomerManagedStatusRequest$ = UpdateReputationEntityCustomerManagedStatusRequest$;
const UpdateReputationEntityCustomerManagedStatusResponse$ = [3, n0, _URECMSRp,
    0,
    [],
    []
];
exports.UpdateReputationEntityCustomerManagedStatusResponse$ = UpdateReputationEntityCustomerManagedStatusResponse$;
const UpdateReputationEntityPolicyRequest$ = [3, n0, _UREPR,
    0,
    [_RET, _RER, _REP],
    [[0, 1], [0, 1], 0], 3
];
exports.UpdateReputationEntityPolicyRequest$ = UpdateReputationEntityPolicyRequest$;
const UpdateReputationEntityPolicyResponse$ = [3, n0, _UREPRp,
    0,
    [],
    []
];
exports.UpdateReputationEntityPolicyResponse$ = UpdateReputationEntityPolicyResponse$;
const VdmAttributes$ = [3, n0, _VA,
    0,
    [_VE, _DAa, _GA],
    [0, () => DashboardAttributes$, () => GuardianAttributes$], 1
];
exports.VdmAttributes$ = VdmAttributes$;
const VdmOptions$ = [3, n0, _VO,
    0,
    [_DOa, _GO],
    [() => DashboardOptions$, () => GuardianOptions$]
];
exports.VdmOptions$ = VdmOptions$;
const VerificationInfo$ = [3, n0, _VI,
    0,
    [_LCT, _LST, _ETr, _SOAR],
    [4, 4, 0, () => SOARecord$]
];
exports.VerificationInfo$ = VerificationInfo$;
const VolumeStatistics$ = [3, n0, _VS,
    0,
    [_IRC, _SRC, _PI, _PSr],
    [1, 1, 1, 1]
];
exports.VolumeStatistics$ = VolumeStatistics$;
var AdditionalContactEmailAddresses = [1, n0, _ACEAd,
    8, [() => AdditionalContactEmailAddress,
        0]
];
var AttachmentList = [1, n0, _AL,
    0, () => Attachment$
];
var BatchGetMetricDataQueries = [1, n0, _BGMDQa,
    0, () => BatchGetMetricDataQuery$
];
var BlacklistEntries = [1, n0, _BEl,
    0, () => BlacklistEntry$
];
var BlacklistItemNames = 64 | 0;
var BulkEmailEntryList = [1, n0, _BEEL,
    0, () => BulkEmailEntry$
];
var BulkEmailEntryResultList = [1, n0, _BEERL,
    0, () => BulkEmailEntryResult$
];
var CloudWatchDimensionConfigurations = [1, n0, _CWDCl,
    0, () => CloudWatchDimensionConfiguration$
];
var ConfigurationSetNameList = 64 | 0;
var CustomVerificationEmailTemplatesList = [1, n0, _CVETL,
    0, () => CustomVerificationEmailTemplateMetadata$
];
var DailyVolumes = [1, n0, _DVa,
    0, () => DailyVolume$
];
var DedicatedIpList = [1, n0, _DIL,
    0, () => DedicatedIp$
];
var DeliverabilityTestReports = [1, n0, _DTReli,
    0, () => DeliverabilityTestReport$
];
var DnsTokenList = 64 | 0;
var DomainDeliverabilityCampaignList = [1, n0, _DDCL,
    0, () => DomainDeliverabilityCampaign$
];
var DomainDeliverabilityTrackingOptions = [1, n0, _DDTOo,
    0, () => DomainDeliverabilityTrackingOption$
];
var DomainIspPlacements = [1, n0, _DIP,
    0, () => DomainIspPlacement$
];
var EmailAddressFilterList = [1, n0, _EAFL,
    0, [() => InsightsEmailAddress,
        0]
];
var EmailAddressList = 64 | 0;
var EmailInsightsList = [1, n0, _EIL,
    0, [() => EmailInsights$,
        0]
];
var EmailSubjectFilterList = [1, n0, _ESFL,
    0, [() => EmailSubject,
        0]
];
var EmailTemplateMetadataList = [1, n0, _ETML,
    0, () => EmailTemplateMetadata$
];
var Esps = 64 | 0;
var EventDestinations = [1, n0, _EDven,
    0, () => EventDestination$
];
var EventTypes = 64 | 0;
var ExportDimensionValue = 64 | 0;
var ExportJobSummaryList = [1, n0, _EJSL,
    0, () => ExportJobSummary$
];
var ExportMetrics = [1, n0, _EMxp,
    0, () => ExportMetric$
];
var IdentityInfoList = [1, n0, _IIL,
    0, () => IdentityInfo$
];
var ImportJobSummaryList = [1, n0, _IJSL,
    0, () => ImportJobSummary$
];
var InsightsEvents = [1, n0, _IEn,
    0, () => InsightsEvent$
];
var IpList = 64 | 0;
var IspFilterList = 64 | 0;
var IspNameList = 64 | 0;
var IspPlacements = [1, n0, _IPs,
    0, () => IspPlacement$
];
var LastDeliveryEventList = 64 | 0;
var LastEngagementEventList = 64 | 0;
var ListOfContactLists = [1, n0, _LOCL,
    0, () => ContactList$
];
var ListOfContacts = [1, n0, _LOC,
    0, () => Contact$
];
var ListOfDedicatedIpPools = 64 | 0;
var MessageHeaderList = [1, n0, _MHL,
    0, () => MessageHeader$
];
var MessageTagList = [1, n0, _MTL,
    0, () => MessageTag$
];
var MetricDataErrorList = [1, n0, _MDEL,
    0, () => MetricDataError$
];
var MetricDataResultList = [1, n0, _MDRL,
    0, () => MetricDataResult$
];
var MetricValueList = 64 | 1;
var MultiRegionEndpoints = [1, n0, _MRE,
    0, () => MultiRegionEndpoint$
];
var RecommendationsList = [1, n0, _RL,
    0, () => Recommendation$
];
var Regions = 64 | 0;
var ReputationEntitiesList = [1, n0, _REL,
    0, () => ReputationEntity$
];
var ResourceTenantMetadataList = [1, n0, _RTML,
    0, () => ResourceTenantMetadata$
];
var Routes = [1, n0, _Ro,
    0, () => Route$
];
var RoutesDetails = [1, n0, _RDo,
    0, () => RouteDetails$
];
var SuppressedDestinationSummaries = [1, n0, _SDS,
    0, () => SuppressedDestinationSummary$
];
var SuppressionListReasons = 64 | 0;
var TagKeyList = 64 | 0;
var TagList = [1, n0, _TL,
    0, () => Tag$
];
var TenantInfoList = [1, n0, _TIL,
    0, () => TenantInfo$
];
var TenantResourceList = [1, n0, _TRL,
    0, () => TenantResource$
];
var TimestampList = 64 | 4;
var TopicPreferenceList = [1, n0, _TPL,
    0, () => TopicPreference$
];
var Topics = [1, n0, _To,
    0, () => Topic$
];
var BlacklistReport = [2, n0, _BR,
    0, 0, () => BlacklistEntries
];
var Dimensions = 128 | 0;
var ExportDimensions = [2, n0, _EDxp,
    0, 0, 64 | 0
];
var ListRecommendationsFilter = 128 | 0;
var ListTenantResourcesFilter = 128 | 0;
var PolicyMap = 128 | 0;
var ReputationEntityFilter = 128 | 0;
exports.BatchGetMetricData$ = [9, n0, _BGMD,
    { [_h]: ["POST", "/v2/email/metrics/batch", 200] }, () => BatchGetMetricDataRequest$, () => BatchGetMetricDataResponse$
];
exports.CancelExportJob$ = [9, n0, _CEJ,
    { [_h]: ["PUT", "/v2/email/export-jobs/{JobId}/cancel", 200] }, () => CancelExportJobRequest$, () => CancelExportJobResponse$
];
exports.CreateConfigurationSet$ = [9, n0, _CCS,
    { [_h]: ["POST", "/v2/email/configuration-sets", 200] }, () => CreateConfigurationSetRequest$, () => CreateConfigurationSetResponse$
];
exports.CreateConfigurationSetEventDestination$ = [9, n0, _CCSED,
    { [_h]: ["POST", "/v2/email/configuration-sets/{ConfigurationSetName}/event-destinations", 200] }, () => CreateConfigurationSetEventDestinationRequest$, () => CreateConfigurationSetEventDestinationResponse$
];
exports.CreateContact$ = [9, n0, _CC,
    { [_h]: ["POST", "/v2/email/contact-lists/{ContactListName}/contacts", 200] }, () => CreateContactRequest$, () => CreateContactResponse$
];
exports.CreateContactList$ = [9, n0, _CCL,
    { [_h]: ["POST", "/v2/email/contact-lists", 200] }, () => CreateContactListRequest$, () => CreateContactListResponse$
];
exports.CreateCustomVerificationEmailTemplate$ = [9, n0, _CCVET,
    { [_h]: ["POST", "/v2/email/custom-verification-email-templates", 200] }, () => CreateCustomVerificationEmailTemplateRequest$, () => CreateCustomVerificationEmailTemplateResponse$
];
exports.CreateDedicatedIpPool$ = [9, n0, _CDIP,
    { [_h]: ["POST", "/v2/email/dedicated-ip-pools", 200] }, () => CreateDedicatedIpPoolRequest$, () => CreateDedicatedIpPoolResponse$
];
exports.CreateDeliverabilityTestReport$ = [9, n0, _CDTR,
    { [_h]: ["POST", "/v2/email/deliverability-dashboard/test", 200] }, () => CreateDeliverabilityTestReportRequest$, () => CreateDeliverabilityTestReportResponse$
];
exports.CreateEmailIdentity$ = [9, n0, _CEI,
    { [_h]: ["POST", "/v2/email/identities", 200] }, () => CreateEmailIdentityRequest$, () => CreateEmailIdentityResponse$
];
exports.CreateEmailIdentityPolicy$ = [9, n0, _CEIP,
    { [_h]: ["POST", "/v2/email/identities/{EmailIdentity}/policies/{PolicyName}", 200] }, () => CreateEmailIdentityPolicyRequest$, () => CreateEmailIdentityPolicyResponse$
];
exports.CreateEmailTemplate$ = [9, n0, _CET,
    { [_h]: ["POST", "/v2/email/templates", 200] }, () => CreateEmailTemplateRequest$, () => CreateEmailTemplateResponse$
];
exports.CreateExportJob$ = [9, n0, _CEJr,
    { [_h]: ["POST", "/v2/email/export-jobs", 200] }, () => CreateExportJobRequest$, () => CreateExportJobResponse$
];
exports.CreateImportJob$ = [9, n0, _CIJ,
    { [_h]: ["POST", "/v2/email/import-jobs", 200] }, () => CreateImportJobRequest$, () => CreateImportJobResponse$
];
exports.CreateMultiRegionEndpoint$ = [9, n0, _CMRE,
    { [_h]: ["POST", "/v2/email/multi-region-endpoints", 200] }, () => CreateMultiRegionEndpointRequest$, () => CreateMultiRegionEndpointResponse$
];
exports.CreateTenant$ = [9, n0, _CTre,
    { [_h]: ["POST", "/v2/email/tenants", 200] }, () => CreateTenantRequest$, () => CreateTenantResponse$
];
exports.CreateTenantResourceAssociation$ = [9, n0, _CTRA,
    { [_h]: ["POST", "/v2/email/tenants/resources", 200] }, () => CreateTenantResourceAssociationRequest$, () => CreateTenantResourceAssociationResponse$
];
exports.DeleteConfigurationSet$ = [9, n0, _DCS,
    { [_h]: ["DELETE", "/v2/email/configuration-sets/{ConfigurationSetName}", 200] }, () => DeleteConfigurationSetRequest$, () => DeleteConfigurationSetResponse$
];
exports.DeleteConfigurationSetEventDestination$ = [9, n0, _DCSED,
    { [_h]: ["DELETE", "/v2/email/configuration-sets/{ConfigurationSetName}/event-destinations/{EventDestinationName}", 200] }, () => DeleteConfigurationSetEventDestinationRequest$, () => DeleteConfigurationSetEventDestinationResponse$
];
exports.DeleteContact$ = [9, n0, _DCel,
    { [_h]: ["DELETE", "/v2/email/contact-lists/{ContactListName}/contacts/{EmailAddress}", 200] }, () => DeleteContactRequest$, () => DeleteContactResponse$
];
exports.DeleteContactList$ = [9, n0, _DCL,
    { [_h]: ["DELETE", "/v2/email/contact-lists/{ContactListName}", 200] }, () => DeleteContactListRequest$, () => DeleteContactListResponse$
];
exports.DeleteCustomVerificationEmailTemplate$ = [9, n0, _DCVET,
    { [_h]: ["DELETE", "/v2/email/custom-verification-email-templates/{TemplateName}", 200] }, () => DeleteCustomVerificationEmailTemplateRequest$, () => DeleteCustomVerificationEmailTemplateResponse$
];
exports.DeleteDedicatedIpPool$ = [9, n0, _DDIP,
    { [_h]: ["DELETE", "/v2/email/dedicated-ip-pools/{PoolName}", 200] }, () => DeleteDedicatedIpPoolRequest$, () => DeleteDedicatedIpPoolResponse$
];
exports.DeleteEmailIdentity$ = [9, n0, _DEI,
    { [_h]: ["DELETE", "/v2/email/identities/{EmailIdentity}", 200] }, () => DeleteEmailIdentityRequest$, () => DeleteEmailIdentityResponse$
];
exports.DeleteEmailIdentityPolicy$ = [9, n0, _DEIP,
    { [_h]: ["DELETE", "/v2/email/identities/{EmailIdentity}/policies/{PolicyName}", 200] }, () => DeleteEmailIdentityPolicyRequest$, () => DeleteEmailIdentityPolicyResponse$
];
exports.DeleteEmailTemplate$ = [9, n0, _DETe,
    { [_h]: ["DELETE", "/v2/email/templates/{TemplateName}", 200] }, () => DeleteEmailTemplateRequest$, () => DeleteEmailTemplateResponse$
];
exports.DeleteMultiRegionEndpoint$ = [9, n0, _DMRE,
    { [_h]: ["DELETE", "/v2/email/multi-region-endpoints/{EndpointName}", 200] }, () => DeleteMultiRegionEndpointRequest$, () => DeleteMultiRegionEndpointResponse$
];
exports.DeleteSuppressedDestination$ = [9, n0, _DSD,
    { [_h]: ["DELETE", "/v2/email/suppression/addresses/{EmailAddress}", 200] }, () => DeleteSuppressedDestinationRequest$, () => DeleteSuppressedDestinationResponse$
];
exports.DeleteTenant$ = [9, n0, _DTe,
    { [_h]: ["POST", "/v2/email/tenants/delete", 200] }, () => DeleteTenantRequest$, () => DeleteTenantResponse$
];
exports.DeleteTenantResourceAssociation$ = [9, n0, _DTRA,
    { [_h]: ["POST", "/v2/email/tenants/resources/delete", 200] }, () => DeleteTenantResourceAssociationRequest$, () => DeleteTenantResourceAssociationResponse$
];
exports.GetAccount$ = [9, n0, _GAe,
    { [_h]: ["GET", "/v2/email/account", 200] }, () => GetAccountRequest$, () => GetAccountResponse$
];
exports.GetBlacklistReports$ = [9, n0, _GBR,
    { [_h]: ["GET", "/v2/email/deliverability-dashboard/blacklist-report", 200] }, () => GetBlacklistReportsRequest$, () => GetBlacklistReportsResponse$
];
exports.GetConfigurationSet$ = [9, n0, _GCS,
    { [_h]: ["GET", "/v2/email/configuration-sets/{ConfigurationSetName}", 200] }, () => GetConfigurationSetRequest$, () => GetConfigurationSetResponse$
];
exports.GetConfigurationSetEventDestinations$ = [9, n0, _GCSED,
    { [_h]: ["GET", "/v2/email/configuration-sets/{ConfigurationSetName}/event-destinations", 200] }, () => GetConfigurationSetEventDestinationsRequest$, () => GetConfigurationSetEventDestinationsResponse$
];
exports.GetContact$ = [9, n0, _GC,
    { [_h]: ["GET", "/v2/email/contact-lists/{ContactListName}/contacts/{EmailAddress}", 200] }, () => GetContactRequest$, () => GetContactResponse$
];
exports.GetContactList$ = [9, n0, _GCL,
    { [_h]: ["GET", "/v2/email/contact-lists/{ContactListName}", 200] }, () => GetContactListRequest$, () => GetContactListResponse$
];
exports.GetCustomVerificationEmailTemplate$ = [9, n0, _GCVET,
    { [_h]: ["GET", "/v2/email/custom-verification-email-templates/{TemplateName}", 200] }, () => GetCustomVerificationEmailTemplateRequest$, () => GetCustomVerificationEmailTemplateResponse$
];
exports.GetDedicatedIp$ = [9, n0, _GDI,
    { [_h]: ["GET", "/v2/email/dedicated-ips/{Ip}", 200] }, () => GetDedicatedIpRequest$, () => GetDedicatedIpResponse$
];
exports.GetDedicatedIpPool$ = [9, n0, _GDIP,
    { [_h]: ["GET", "/v2/email/dedicated-ip-pools/{PoolName}", 200] }, () => GetDedicatedIpPoolRequest$, () => GetDedicatedIpPoolResponse$
];
exports.GetDedicatedIps$ = [9, n0, _GDIe,
    { [_h]: ["GET", "/v2/email/dedicated-ips", 200] }, () => GetDedicatedIpsRequest$, () => GetDedicatedIpsResponse$
];
exports.GetDeliverabilityDashboardOptions$ = [9, n0, _GDDO,
    { [_h]: ["GET", "/v2/email/deliverability-dashboard", 200] }, () => GetDeliverabilityDashboardOptionsRequest$, () => GetDeliverabilityDashboardOptionsResponse$
];
exports.GetDeliverabilityTestReport$ = [9, n0, _GDTR,
    { [_h]: ["GET", "/v2/email/deliverability-dashboard/test-reports/{ReportId}", 200] }, () => GetDeliverabilityTestReportRequest$, () => GetDeliverabilityTestReportResponse$
];
exports.GetDomainDeliverabilityCampaign$ = [9, n0, _GDDC,
    { [_h]: ["GET", "/v2/email/deliverability-dashboard/campaigns/{CampaignId}", 200] }, () => GetDomainDeliverabilityCampaignRequest$, () => GetDomainDeliverabilityCampaignResponse$
];
exports.GetDomainStatisticsReport$ = [9, n0, _GDSR,
    { [_h]: ["GET", "/v2/email/deliverability-dashboard/statistics-report/{Domain}", 200] }, () => GetDomainStatisticsReportRequest$, () => GetDomainStatisticsReportResponse$
];
exports.GetEmailAddressInsights$ = [9, n0, _GEAI,
    { [_h]: ["POST", "/v2/email/email-address-insights", 200] }, () => GetEmailAddressInsightsRequest$, () => GetEmailAddressInsightsResponse$
];
exports.GetEmailIdentity$ = [9, n0, _GEI,
    { [_h]: ["GET", "/v2/email/identities/{EmailIdentity}", 200] }, () => GetEmailIdentityRequest$, () => GetEmailIdentityResponse$
];
exports.GetEmailIdentityPolicies$ = [9, n0, _GEIP,
    { [_h]: ["GET", "/v2/email/identities/{EmailIdentity}/policies", 200] }, () => GetEmailIdentityPoliciesRequest$, () => GetEmailIdentityPoliciesResponse$
];
exports.GetEmailTemplate$ = [9, n0, _GET,
    { [_h]: ["GET", "/v2/email/templates/{TemplateName}", 200] }, () => GetEmailTemplateRequest$, () => GetEmailTemplateResponse$
];
exports.GetExportJob$ = [9, n0, _GEJ,
    { [_h]: ["GET", "/v2/email/export-jobs/{JobId}", 200] }, () => GetExportJobRequest$, () => GetExportJobResponse$
];
exports.GetImportJob$ = [9, n0, _GIJ,
    { [_h]: ["GET", "/v2/email/import-jobs/{JobId}", 200] }, () => GetImportJobRequest$, () => GetImportJobResponse$
];
exports.GetMessageInsights$ = [9, n0, _GMI,
    { [_h]: ["GET", "/v2/email/insights/{MessageId}", 200] }, () => GetMessageInsightsRequest$, () => GetMessageInsightsResponse$
];
exports.GetMultiRegionEndpoint$ = [9, n0, _GMRE,
    { [_h]: ["GET", "/v2/email/multi-region-endpoints/{EndpointName}", 200] }, () => GetMultiRegionEndpointRequest$, () => GetMultiRegionEndpointResponse$
];
exports.GetReputationEntity$ = [9, n0, _GRE,
    { [_h]: ["GET", "/v2/email/reputation/entities/{ReputationEntityType}/{ReputationEntityReference}", 200] }, () => GetReputationEntityRequest$, () => GetReputationEntityResponse$
];
exports.GetSuppressedDestination$ = [9, n0, _GSD,
    { [_h]: ["GET", "/v2/email/suppression/addresses/{EmailAddress}", 200] }, () => GetSuppressedDestinationRequest$, () => GetSuppressedDestinationResponse$
];
exports.GetTenant$ = [9, n0, _GT,
    { [_h]: ["POST", "/v2/email/tenants/get", 200] }, () => GetTenantRequest$, () => GetTenantResponse$
];
exports.ListConfigurationSets$ = [9, n0, _LCS,
    { [_h]: ["GET", "/v2/email/configuration-sets", 200] }, () => ListConfigurationSetsRequest$, () => ListConfigurationSetsResponse$
];
exports.ListContactLists$ = [9, n0, _LCL,
    { [_h]: ["GET", "/v2/email/contact-lists", 200] }, () => ListContactListsRequest$, () => ListContactListsResponse$
];
exports.ListContacts$ = [9, n0, _LC,
    { [_h]: ["POST", "/v2/email/contact-lists/{ContactListName}/contacts/list", 200] }, () => ListContactsRequest$, () => ListContactsResponse$
];
exports.ListCustomVerificationEmailTemplates$ = [9, n0, _LCVET,
    { [_h]: ["GET", "/v2/email/custom-verification-email-templates", 200] }, () => ListCustomVerificationEmailTemplatesRequest$, () => ListCustomVerificationEmailTemplatesResponse$
];
exports.ListDedicatedIpPools$ = [9, n0, _LDIP,
    { [_h]: ["GET", "/v2/email/dedicated-ip-pools", 200] }, () => ListDedicatedIpPoolsRequest$, () => ListDedicatedIpPoolsResponse$
];
exports.ListDeliverabilityTestReports$ = [9, n0, _LDTR,
    { [_h]: ["GET", "/v2/email/deliverability-dashboard/test-reports", 200] }, () => ListDeliverabilityTestReportsRequest$, () => ListDeliverabilityTestReportsResponse$
];
exports.ListDomainDeliverabilityCampaigns$ = [9, n0, _LDDC,
    { [_h]: ["GET", "/v2/email/deliverability-dashboard/domains/{SubscribedDomain}/campaigns", 200] }, () => ListDomainDeliverabilityCampaignsRequest$, () => ListDomainDeliverabilityCampaignsResponse$
];
exports.ListEmailIdentities$ = [9, n0, _LEI,
    { [_h]: ["GET", "/v2/email/identities", 200] }, () => ListEmailIdentitiesRequest$, () => ListEmailIdentitiesResponse$
];
exports.ListEmailTemplates$ = [9, n0, _LET,
    { [_h]: ["GET", "/v2/email/templates", 200] }, () => ListEmailTemplatesRequest$, () => ListEmailTemplatesResponse$
];
exports.ListExportJobs$ = [9, n0, _LEJ,
    { [_h]: ["POST", "/v2/email/list-export-jobs", 200] }, () => ListExportJobsRequest$, () => ListExportJobsResponse$
];
exports.ListImportJobs$ = [9, n0, _LIJ,
    { [_h]: ["POST", "/v2/email/import-jobs/list", 200] }, () => ListImportJobsRequest$, () => ListImportJobsResponse$
];
exports.ListMultiRegionEndpoints$ = [9, n0, _LMRE,
    { [_h]: ["GET", "/v2/email/multi-region-endpoints", 200] }, () => ListMultiRegionEndpointsRequest$, () => ListMultiRegionEndpointsResponse$
];
exports.ListRecommendations$ = [9, n0, _LR,
    { [_h]: ["POST", "/v2/email/vdm/recommendations", 200] }, () => ListRecommendationsRequest$, () => ListRecommendationsResponse$
];
exports.ListReputationEntities$ = [9, n0, _LRE,
    { [_h]: ["POST", "/v2/email/reputation/entities", 200] }, () => ListReputationEntitiesRequest$, () => ListReputationEntitiesResponse$
];
exports.ListResourceTenants$ = [9, n0, _LRT,
    { [_h]: ["POST", "/v2/email/resources/tenants/list", 200] }, () => ListResourceTenantsRequest$, () => ListResourceTenantsResponse$
];
exports.ListSuppressedDestinations$ = [9, n0, _LSD,
    { [_h]: ["GET", "/v2/email/suppression/addresses", 200] }, () => ListSuppressedDestinationsRequest$, () => ListSuppressedDestinationsResponse$
];
exports.ListTagsForResource$ = [9, n0, _LTFR,
    { [_h]: ["GET", "/v2/email/tags", 200] }, () => ListTagsForResourceRequest$, () => ListTagsForResourceResponse$
];
exports.ListTenantResources$ = [9, n0, _LTRis,
    { [_h]: ["POST", "/v2/email/tenants/resources/list", 200] }, () => ListTenantResourcesRequest$, () => ListTenantResourcesResponse$
];
exports.ListTenants$ = [9, n0, _LTi,
    { [_h]: ["POST", "/v2/email/tenants/list", 200] }, () => ListTenantsRequest$, () => ListTenantsResponse$
];
exports.PutAccountDedicatedIpWarmupAttributes$ = [9, n0, _PADIWA,
    { [_h]: ["PUT", "/v2/email/account/dedicated-ips/warmup", 200] }, () => PutAccountDedicatedIpWarmupAttributesRequest$, () => PutAccountDedicatedIpWarmupAttributesResponse$
];
exports.PutAccountDetails$ = [9, n0, _PAD,
    { [_h]: ["POST", "/v2/email/account/details", 200] }, () => PutAccountDetailsRequest$, () => PutAccountDetailsResponse$
];
exports.PutAccountSendingAttributes$ = [9, n0, _PASA,
    { [_h]: ["PUT", "/v2/email/account/sending", 200] }, () => PutAccountSendingAttributesRequest$, () => PutAccountSendingAttributesResponse$
];
exports.PutAccountSuppressionAttributes$ = [9, n0, _PASAu,
    { [_h]: ["PUT", "/v2/email/account/suppression", 200] }, () => PutAccountSuppressionAttributesRequest$, () => PutAccountSuppressionAttributesResponse$
];
exports.PutAccountVdmAttributes$ = [9, n0, _PAVA,
    { [_h]: ["PUT", "/v2/email/account/vdm", 200] }, () => PutAccountVdmAttributesRequest$, () => PutAccountVdmAttributesResponse$
];
exports.PutConfigurationSetArchivingOptions$ = [9, n0, _PCSAO,
    { [_h]: ["PUT", "/v2/email/configuration-sets/{ConfigurationSetName}/archiving-options", 200] }, () => PutConfigurationSetArchivingOptionsRequest$, () => PutConfigurationSetArchivingOptionsResponse$
];
exports.PutConfigurationSetDeliveryOptions$ = [9, n0, _PCSDO,
    { [_h]: ["PUT", "/v2/email/configuration-sets/{ConfigurationSetName}/delivery-options", 200] }, () => PutConfigurationSetDeliveryOptionsRequest$, () => PutConfigurationSetDeliveryOptionsResponse$
];
exports.PutConfigurationSetReputationOptions$ = [9, n0, _PCSRO,
    { [_h]: ["PUT", "/v2/email/configuration-sets/{ConfigurationSetName}/reputation-options", 200] }, () => PutConfigurationSetReputationOptionsRequest$, () => PutConfigurationSetReputationOptionsResponse$
];
exports.PutConfigurationSetSendingOptions$ = [9, n0, _PCSSO,
    { [_h]: ["PUT", "/v2/email/configuration-sets/{ConfigurationSetName}/sending", 200] }, () => PutConfigurationSetSendingOptionsRequest$, () => PutConfigurationSetSendingOptionsResponse$
];
exports.PutConfigurationSetSuppressionOptions$ = [9, n0, _PCSSOu,
    { [_h]: ["PUT", "/v2/email/configuration-sets/{ConfigurationSetName}/suppression-options", 200] }, () => PutConfigurationSetSuppressionOptionsRequest$, () => PutConfigurationSetSuppressionOptionsResponse$
];
exports.PutConfigurationSetTrackingOptions$ = [9, n0, _PCSTO,
    { [_h]: ["PUT", "/v2/email/configuration-sets/{ConfigurationSetName}/tracking-options", 200] }, () => PutConfigurationSetTrackingOptionsRequest$, () => PutConfigurationSetTrackingOptionsResponse$
];
exports.PutConfigurationSetVdmOptions$ = [9, n0, _PCSVO,
    { [_h]: ["PUT", "/v2/email/configuration-sets/{ConfigurationSetName}/vdm-options", 200] }, () => PutConfigurationSetVdmOptionsRequest$, () => PutConfigurationSetVdmOptionsResponse$
];
exports.PutDedicatedIpInPool$ = [9, n0, _PDIIP,
    { [_h]: ["PUT", "/v2/email/dedicated-ips/{Ip}/pool", 200] }, () => PutDedicatedIpInPoolRequest$, () => PutDedicatedIpInPoolResponse$
];
exports.PutDedicatedIpPoolScalingAttributes$ = [9, n0, _PDIPSA,
    { [_h]: ["PUT", "/v2/email/dedicated-ip-pools/{PoolName}/scaling", 200] }, () => PutDedicatedIpPoolScalingAttributesRequest$, () => PutDedicatedIpPoolScalingAttributesResponse$
];
exports.PutDedicatedIpWarmupAttributes$ = [9, n0, _PDIWA,
    { [_h]: ["PUT", "/v2/email/dedicated-ips/{Ip}/warmup", 200] }, () => PutDedicatedIpWarmupAttributesRequest$, () => PutDedicatedIpWarmupAttributesResponse$
];
exports.PutDeliverabilityDashboardOption$ = [9, n0, _PDDO,
    { [_h]: ["PUT", "/v2/email/deliverability-dashboard", 200] }, () => PutDeliverabilityDashboardOptionRequest$, () => PutDeliverabilityDashboardOptionResponse$
];
exports.PutEmailIdentityConfigurationSetAttributes$ = [9, n0, _PEICSA,
    { [_h]: ["PUT", "/v2/email/identities/{EmailIdentity}/configuration-set", 200] }, () => PutEmailIdentityConfigurationSetAttributesRequest$, () => PutEmailIdentityConfigurationSetAttributesResponse$
];
exports.PutEmailIdentityDkimAttributes$ = [9, n0, _PEIDA,
    { [_h]: ["PUT", "/v2/email/identities/{EmailIdentity}/dkim", 200] }, () => PutEmailIdentityDkimAttributesRequest$, () => PutEmailIdentityDkimAttributesResponse$
];
exports.PutEmailIdentityDkimSigningAttributes$ = [9, n0, _PEIDSA,
    { [_h]: ["PUT", "/v2/email/identities/{EmailIdentity}/dkim/signing", 200] }, () => PutEmailIdentityDkimSigningAttributesRequest$, () => PutEmailIdentityDkimSigningAttributesResponse$
];
exports.PutEmailIdentityFeedbackAttributes$ = [9, n0, _PEIFA,
    { [_h]: ["PUT", "/v2/email/identities/{EmailIdentity}/feedback", 200] }, () => PutEmailIdentityFeedbackAttributesRequest$, () => PutEmailIdentityFeedbackAttributesResponse$
];
exports.PutEmailIdentityMailFromAttributes$ = [9, n0, _PEIMFA,
    { [_h]: ["PUT", "/v2/email/identities/{EmailIdentity}/mail-from", 200] }, () => PutEmailIdentityMailFromAttributesRequest$, () => PutEmailIdentityMailFromAttributesResponse$
];
exports.PutSuppressedDestination$ = [9, n0, _PSD,
    { [_h]: ["PUT", "/v2/email/suppression/addresses", 200] }, () => PutSuppressedDestinationRequest$, () => PutSuppressedDestinationResponse$
];
exports.PutTenantSuppressionAttributes$ = [9, n0, _PTSA,
    { [_h]: ["POST", "/v2/email/tenant/suppression", 200] }, () => PutTenantSuppressionAttributesRequest$, () => PutTenantSuppressionAttributesResponse$
];
exports.SendBulkEmail$ = [9, n0, _SBE,
    { [_h]: ["POST", "/v2/email/outbound-bulk-emails", 200] }, () => SendBulkEmailRequest$, () => SendBulkEmailResponse$
];
exports.SendCustomVerificationEmail$ = [9, n0, _SCVE,
    { [_h]: ["POST", "/v2/email/outbound-custom-verification-emails", 200] }, () => SendCustomVerificationEmailRequest$, () => SendCustomVerificationEmailResponse$
];
exports.SendEmail$ = [9, n0, _SEen,
    { [_h]: ["POST", "/v2/email/outbound-emails", 200] }, () => SendEmailRequest$, () => SendEmailResponse$
];
exports.TagResource$ = [9, n0, _TRa,
    { [_h]: ["POST", "/v2/email/tags", 200] }, () => TagResourceRequest$, () => TagResourceResponse$
];
exports.TestRenderEmailTemplate$ = [9, n0, _TRET,
    { [_h]: ["POST", "/v2/email/templates/{TemplateName}/render", 200] }, () => TestRenderEmailTemplateRequest$, () => TestRenderEmailTemplateResponse$
];
exports.UntagResource$ = [9, n0, _UR,
    { [_h]: ["DELETE", "/v2/email/tags", 200] }, () => UntagResourceRequest$, () => UntagResourceResponse$
];
exports.UpdateConfigurationSetEventDestination$ = [9, n0, _UCSED,
    { [_h]: ["PUT", "/v2/email/configuration-sets/{ConfigurationSetName}/event-destinations/{EventDestinationName}", 200] }, () => UpdateConfigurationSetEventDestinationRequest$, () => UpdateConfigurationSetEventDestinationResponse$
];
exports.UpdateContact$ = [9, n0, _UC,
    { [_h]: ["PUT", "/v2/email/contact-lists/{ContactListName}/contacts/{EmailAddress}", 200] }, () => UpdateContactRequest$, () => UpdateContactResponse$
];
exports.UpdateContactList$ = [9, n0, _UCL,
    { [_h]: ["PUT", "/v2/email/contact-lists/{ContactListName}", 200] }, () => UpdateContactListRequest$, () => UpdateContactListResponse$
];
exports.UpdateCustomVerificationEmailTemplate$ = [9, n0, _UCVET,
    { [_h]: ["PUT", "/v2/email/custom-verification-email-templates/{TemplateName}", 200] }, () => UpdateCustomVerificationEmailTemplateRequest$, () => UpdateCustomVerificationEmailTemplateResponse$
];
exports.UpdateEmailIdentityPolicy$ = [9, n0, _UEIP,
    { [_h]: ["PUT", "/v2/email/identities/{EmailIdentity}/policies/{PolicyName}", 200] }, () => UpdateEmailIdentityPolicyRequest$, () => UpdateEmailIdentityPolicyResponse$
];
exports.UpdateEmailTemplate$ = [9, n0, _UET,
    { [_h]: ["PUT", "/v2/email/templates/{TemplateName}", 200] }, () => UpdateEmailTemplateRequest$, () => UpdateEmailTemplateResponse$
];
exports.UpdateReputationEntityCustomerManagedStatus$ = [9, n0, _URECMS,
    { [_h]: ["PUT", "/v2/email/reputation/entities/{ReputationEntityType}/{ReputationEntityReference}/customer-managed-status", 200] }, () => UpdateReputationEntityCustomerManagedStatusRequest$, () => UpdateReputationEntityCustomerManagedStatusResponse$
];
exports.UpdateReputationEntityPolicy$ = [9, n0, _UREP,
    { [_h]: ["PUT", "/v2/email/reputation/entities/{ReputationEntityType}/{ReputationEntityReference}/policy", 200] }, () => UpdateReputationEntityPolicyRequest$, () => UpdateReputationEntityPolicyResponse$
];
